<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Tools for Accessing and Working with the Gene Expression Omnibus (GEO) • geokit</title>
<script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Tools for Accessing and Working with the Gene Expression Omnibus (GEO)">
<meta name="description" content="Provides a tidy and convenient R interface to the NCBI Gene Expression Omnibus (GEO) database. Functions are included to query, download, and parse GEO metadata and expression data, making it easier to integrate GEO datasets into downstream analyses and workflows.">
<meta property="og:description" content="Provides a tidy and convenient R interface to the NCBI Gene Expression Omnibus (GEO) database. Functions are included to query, download, and parse GEO metadata and expression data, making it easier to integrate GEO datasets into downstream analyses and workflows.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">geokit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Yunuuuu/geokit/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="geokit">geokit<a class="anchor" aria-label="anchor" href="#geokit"></a>
</h1></div>
<!-- badges: start -->

<p>The goal of <code>geokit</code> is to provide a unified interface for most interactions between R and <a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link">GEO database</a>.</p>
<div class="section level2">
<h2 id="features">Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h2>
<ul>
<li>Low dependency and Consistent design, use <a href="https://github.com/jeroen/curl" class="external-link"><code>curl</code></a> to download all files, and utilize <a href="https://github.com/Rdatatable/data.table" class="external-link"><code>data.table</code></a> to implement all reading and preprocessing process. Reducing the dependencies is the initial purpose of this package since I have experienced several times of code running failure after updating packages when using <a href="https://github.com/seandavi/GEOquery" class="external-link"><code>GEOquery</code></a>.</li>
<li>Provide a searching interface of <a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link">GEO database</a>, in this way, we can filter the searching results using <code>R function</code>.</li>
<li>Provide a downloading interface of <a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link">GEO database</a>, in this way, we can make full use of R to analyze GEO datasets.</li>
<li>Enable mapping bettween GPL id and Bioconductor annotation package.</li>
<li>Provide some useful utils function to work with GEO datasets like <code>parse_gsm_list</code>, <code>parse_pdata</code>, <code>log_trans</code> and <code>geo_show</code>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of <code>geokit</code> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"pak"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"pak"</span>,</span>
<span>        repos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span></span>
<span>            <span class="st">"https://r-lib.github.io/p/pak/devel/%s/%s/%s"</span>,</span>
<span>            <span class="va">.Platform</span><span class="op">$</span><span class="va">pkgType</span>, <span class="fu"><a href="https://rdrr.io/r/base/Version.html" class="external-link">R.Version</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">os</span>, <span class="fu"><a href="https://rdrr.io/r/base/Version.html" class="external-link">R.Version</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">arch</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">"Yunuuuu/geokit"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vignettes">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Yunuuuu/geokit" class="external-link">geokit</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="search-geo-database---geo_search">Search GEO database - <code>geo_search</code>
<a class="anchor" aria-label="anchor" href="#search-geo-database---geo_search"></a>
</h3>
<p>The NCBI uses a search term syntax which can be associated with a specific search field enclosed by a pair of square brackets. So, for instance <code>"Homo sapiens[ORGN]"</code> denotes a search for <code>Homo sapiens</code> in the <code>“Organism”</code> field. Details see <a href="https://www.ncbi.nlm.nih.gov/geo/info/qqtutorial.html" class="external-link uri">https://www.ncbi.nlm.nih.gov/geo/info/qqtutorial.html</a>. We can use the same term to query our desirable results in <code>geo_search</code>. <code>geo_search</code> will parse the searching results and return a <code>data.frame</code> object containing all the records based on the search term. The internal of <code>geo_search</code> is based on <a href="https://github.com/ropensci/rentrez" class="external-link"><code>rentrez</code></a> package, which provides functions working with the <a href="http://www.ncbi.nlm.nih.gov/books/NBK25500/" class="external-link">NCBI Eutils</a> API, so we can utilize <code>NCBI API key</code> to increase the downloading speed, details see <a href="https://docs.ropensci.org/rentrez/articles/rentrez_tutorial.html#rate-limiting-and-api-keys" class="external-link uri">https://docs.ropensci.org/rentrez/articles/rentrez_tutorial.html#rate-limiting-and-api-keys</a>.</p>
<p>Providing we want <strong><em>GSE</em></strong> GEO records related to <strong><em>human diabetes</em></strong>, we can get these records by following code, the returned object is a <code>data.frame</code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diabetes_gse_records</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo_search.html">geo_search</a></span><span class="op">(</span></span>
<span>    <span class="st">"diabetes[ALL] AND Homo sapiens[ORGN] AND GSE[ETYP]"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ■■■■■■■■■■                       500/1690 [484/s] | ETA:  2s</span></span>
<span><span class="co">#&gt; ■■■■■■■■■■■■■■■■■■■              1000/1690 [387/s] | ETA:  2s</span></span>
<span><span class="co">#&gt; Get records from NCBI for 1690 queries in 4.8s</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; → Parsing GEO records</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">diabetes_gse_records</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                                                                                    Title</span></span>
<span><span class="co">#&gt;                                                                                                                   &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:             Coxsackievirus B infection invokes unique cell-type specific responses in primary human pancreatic islets</span></span>
<span><span class="co">#&gt; 2: Expression data from type 2 diabetes mellitus adipose-derived stem cells cultured with basic fibroblast growth factor</span></span>
<span><span class="co">#&gt; 3:                 Engineered vasculature induces functional maturation of pluripotent stem cell-derived islet organoids</span></span>
<span><span class="co">#&gt; 4:                               Recessive TMEM167A variants cause neonatal diabetes, microcephaly and epilepsy syndrome</span></span>
<span><span class="co">#&gt; 5:                   Recessive TMEM167A variants cause neonatal diabetes, microcephaly and epilepsy syndrome [scRNA-seq]</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Summary</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:                                                                                                               Coxsackievirus B (CVB) infection has long been considered an environmental factor precipitating Type 1 diabetes (T1D), an autoimmune disease marked by loss of insulin-producing b cells within pancreatic islets. Previous studies have shown CVB infection negatively impacts islet function and viability but do not report on how virus infection individually affects the multiple cell types present in human primary islets. more...</span></span>
<span><span class="co">#&gt; 2:                  Diabetes affects ASCs characteristics such as: proliferation, differentiation and angiogenic capacity. MicroRNAs are able to target genes involved in vascular remodeling and promote or inhibit structural changes in the vessel wall. Adipose tissue-derived stem cells (ASCs) have the capacity to contribute to vascular remodeling. We used microarrays to detail the global miRNA expression profile underlying cell differentiation and identified distinct classes of up-regulated and down-regulated genes during this process.</span></span>
<span><span class="co">#&gt; 3: Blood vessels play a critical role in pancreatic islet function, yet current methods for deriving islet organoids from human pluripotent stem cells (SC-islets) lack vasculature. We engineered 3D vascularized SC-islet organoids by assembling SC-islet cells, human primary endothelial cells (ECs) and fibroblasts in a non-perfused model and a microfluidic device with perfused vessels. Vasculature improved stimulus-dependent Ca2+ influx into SC-β-cells, a hallmark of β-cell function that is blunted in non-vascularized SC-islets. more...</span></span>
<span><span class="co">#&gt; 4:                                  Understanding the genetic causes of diseases affecting pancreatic β cells and neurons can give insights into pathways essential for both cell types. Microcephaly, epilepsy and diabetes syndrome (MEDS) is a congenital disorder with two known aetiological genes, IER3IP1 and YIPF5. Both genes encode proteins involved in endoplasmic reticulum (ER) to Golgi trafficking.  We used genome sequencing to identify 6 individuals with MEDS caused by biallelic variants in the novel disease gene, TMEM167A. more...</span></span>
<span><span class="co">#&gt; 5:                                  Understanding the genetic causes of diseases affecting pancreatic β cells and neurons can give insights into pathways essential for both cell types. Microcephaly, epilepsy and diabetes syndrome (MEDS) is a congenital disorder with two known aetiological genes, IER3IP1 and YIPF5. Both genes encode proteins involved in endoplasmic reticulum (ER) to Golgi trafficking.  We used genome sequencing to identify 6 individuals with MEDS caused by biallelic variants in the novel disease gene, TMEM167A. more...</span></span>
<span><span class="co">#&gt;                             Organism</span></span>
<span><span class="co">#&gt;                               &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:                      Homo sapiens</span></span>
<span><span class="co">#&gt; 2: Homo sapiens; synthetic construct</span></span>
<span><span class="co">#&gt; 3:                      Homo sapiens</span></span>
<span><span class="co">#&gt; 4:                      Homo sapiens</span></span>
<span><span class="co">#&gt; 5:                      Homo sapiens</span></span>
<span><span class="co">#&gt;                                                  Type</span></span>
<span><span class="co">#&gt;                                                &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1: Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt; 2:                  Non-coding RNA profiling by array</span></span>
<span><span class="co">#&gt; 3: Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt; 4: Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt; 5: Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt;                                                                 FTP download</span></span>
<span><span class="co">#&gt;                                                                       &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1: GEO (MTX, TSV) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE274nnn/GSE274264/</span></span>
<span><span class="co">#&gt; 2:      GEO (CEL) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE283nnn/GSE283040/</span></span>
<span><span class="co">#&gt; 3: GEO (MTX, TSV) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE276nnn/GSE276815/</span></span>
<span><span class="co">#&gt; 4:      GEO (CSV) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE302nnn/GSE302570/</span></span>
<span><span class="co">#&gt; 5: GEO (MTX, TSV) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE302nnn/GSE302421/</span></span>
<span><span class="co">#&gt;           ID SRA Run Selector Project   Contains Datasets Platforms</span></span>
<span><span class="co">#&gt;        &lt;int&gt;           &lt;char&gt;  &lt;char&gt;     &lt;char&gt;   &lt;char&gt;    &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1: 200274264             &lt;NA&gt;    &lt;NA&gt; 10 Samples     &lt;NA&gt;  GPL24676</span></span>
<span><span class="co">#&gt; 2: 200283040             &lt;NA&gt;    &lt;NA&gt; 16 Samples     &lt;NA&gt;  GPL19117</span></span>
<span><span class="co">#&gt; 3: 200276815             &lt;NA&gt;    &lt;NA&gt;  2 Samples     &lt;NA&gt;  GPL24676</span></span>
<span><span class="co">#&gt; 4: 200302570             &lt;NA&gt;    &lt;NA&gt; 40 Samples     &lt;NA&gt;  GPL34284</span></span>
<span><span class="co">#&gt; 5: 200302421             &lt;NA&gt;    &lt;NA&gt;  3 Samples     &lt;NA&gt;  GPL24676</span></span>
<span><span class="co">#&gt;    Series Accession</span></span>
<span><span class="co">#&gt;              &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:        GSE274264</span></span>
<span><span class="co">#&gt; 2:        GSE283040</span></span>
<span><span class="co">#&gt; 3:        GSE276815</span></span>
<span><span class="co">#&gt; 4:        GSE302570</span></span>
<span><span class="co">#&gt; 5:        GSE302421</span></span></code></pre></div>
<p>Then, we can use whatever we’re famaliar to filter the searching results. Providing we want GSE datasets with at least 6 diabetic nephropathy samples containing expression profiling. Here is the example code:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diabetes_nephropathy_gse_records</span> <span class="op">&lt;-</span> <span class="va">diabetes_gse_records</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">mutate</span><span class="op">(</span></span>
<span>        number_of_samples <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_match.html" class="external-link">str_match</a></span><span class="op">(</span></span>
<span>            <span class="va">Contains</span>, <span class="st">"(\\d+) Samples?"</span></span>
<span>        <span class="op">)</span><span class="op">[</span>, <span class="fl">2L</span>, drop <span class="op">=</span> <span class="cn">TRUE</span><span class="op">]</span>,</span>
<span>        number_of_samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">number_of_samples</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">if_any</span><span class="op">(</span></span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Title</span>, <span class="va">Summary</span><span class="op">)</span>,</span>
<span>            <span class="op">~</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">.x</span>, <span class="st">"(?i)diabetes|diabetic"</span><span class="op">)</span></span>
<span>        <span class="op">)</span>,</span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">if_any</span><span class="op">(</span></span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Title</span>, <span class="va">Summary</span><span class="op">)</span>,</span>
<span>            <span class="op">~</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">.x</span>, <span class="st">"(?i)nephropathy"</span><span class="op">)</span></span>
<span>        <span class="op">)</span>,</span>
<span>        <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">Type</span>, <span class="st">"(?i)expression profiling"</span><span class="op">)</span>,</span>
<span>        <span class="va">number_of_samples</span> <span class="op">&gt;=</span> <span class="fl">6L</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">diabetes_nephropathy_gse_records</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                                                                                                Title</span></span>
<span><span class="co">#&gt;                                                                                                                               &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:   Endothelial Kallikrein-Related Peptidase 8 Promotes Diabetic Nephropathy via Reducing SDC4 Expression and Enhancing LIF Release</span></span>
<span><span class="co">#&gt; 2:         Upregulation of FGF13 promotes type 2 diabetic nephropathy by modulating glomerular endothelial mitochondrial homeostasis</span></span>
<span><span class="co">#&gt; 3:                   Sodium Butyrate Ameliorates Renal Tubular Lipid Accumulation Through the PP2A-TFEB axis in Diabetic Nephropathy</span></span>
<span><span class="co">#&gt; 4: Deciphering the Transcriptomic Landscape of Type 2 Diabetes: Insights from Bulk RNA Sequencing and Single-Cell Analysis [RNA-seq]</span></span>
<span><span class="co">#&gt; 5:                                                                  Effect of overexpssion Kallistatin(SERPINA4) in HGC-27 cell line</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Summary</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:                                                                                                                                                                                                                                                                                                                                                       The molecular mechanisms underlying diabetic nephropathy (DN) are poorly defined. We sought to investigate the roles of kallikrein-related peptidases (KLKs) in DN pathogenesis. Screening of renal tissue from diabetic mice revealed KLK8 as the most highly induced gene in KLK family. KLK8 expression was greater in glomerular endothelial cells (GECs) than other glomerular cells in DN patients and diabetic mice. more...</span></span>
<span><span class="co">#&gt; 2:                                                                                           Studies of diabetic glomerular injury raise the possibility of developing useful early biomarkers and therapeutic approaches for the treatment of type 2 diabetic nephropathy (T2DN). In this study, it is found that FGF13 expression is induced in glomerular endothelial cells (GECs) during T2DN progression, and endothelial-specific deletion of Fgf13 potentially alleviates T2DN damage. Fgf13 deficiency restores the expression of Parkin both in the cytosolic, mitochondrial, and nuclear fractions under diabetic conditions, resulting in improved mitochondrial homeostasis and endothelial barrier integrity due to promotion of mitophagy and inhibition of apoptosis. more...</span></span>
<span><span class="co">#&gt; 3:                                                                                                                                                                                                                                                                                                                         Background: Diabetic kidney disease (DKD) is the leading cause of end-stage renal disease worldwide with limited treatment options. The intricate pathogenesis of dysregulated lipid metabolism leading to the development of DKD remains obscure. Lipophagy, which refers to the autophagic degradation of intracellular lipid droplets, has been found to be impaired in DKD, resulting in renal tubule dysfunction and ectopic lipid deposition (ELD). more...</span></span>
<span><span class="co">#&gt; 4:                                                                                                                                                            Type 2 diabetes (T2D) is a chronic metabolic disorder characterized by insulin resistance and relative insulin deficiency. It is a significant public health concern worldwide, with an estimated prevalence of over 422 million individuals affected globally. This number is projected to rise, making diabetes one of the leading causes of morbidity and mortality. It is associated with numerous severe microvascular and macrovascular complications, including retinopathy, nephropathy, cardiovascular diseases, and neuropathy, which substantially impact patients' quality of life and healthcare systems. more...</span></span>
<span><span class="co">#&gt; 5: Kallistatin has been demonstrated to possess inhibitory effects across several malignancies, including hepatocellular carcinoma, gastric cancer and breast cancer. Subsequent evidence has increasingly suggested that KS has pleiotropic roles in modulating a broad spectrum of diseases, including in diabetic nephropathy, idiopathic pulmonary fibrosis and autoimmune uveitis. However, the precise function and molecular mechanisms underlying tumor-induced immune escape attributed to KS remain unclear, necessitating further investigation to determine its role in this context.For this propose, we establish SERPINA4 stably expressed cell line(and control) in HGC-27 cells, and RNA-seq was performed to reveal the trancriptome changes between there two cell lines.</span></span>
<span><span class="co">#&gt;        Organism                                               Type</span></span>
<span><span class="co">#&gt;          &lt;char&gt;                                             &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1: Homo sapiens Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt; 2: Homo sapiens Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt; 3: Homo sapiens Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt; 4: Homo sapiens Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt; 5: Homo sapiens Expression profiling by high throughput sequencing</span></span>
<span><span class="co">#&gt;                                                            FTP download</span></span>
<span><span class="co">#&gt;                                                                  &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE255nnn/GSE255028/</span></span>
<span><span class="co">#&gt; 2: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE192nnn/GSE192889/</span></span>
<span><span class="co">#&gt; 3: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE266nnn/GSE266108/</span></span>
<span><span class="co">#&gt; 4: GEO (TSV) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE280nnn/GSE280402/</span></span>
<span><span class="co">#&gt; 5: GEO (TXT) ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE262nnn/GSE262922/</span></span>
<span><span class="co">#&gt;           ID SRA Run Selector Project   Contains Datasets Platforms</span></span>
<span><span class="co">#&gt;        &lt;int&gt;           &lt;char&gt;  &lt;char&gt;     &lt;char&gt;   &lt;char&gt;    &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1: 200255028             &lt;NA&gt;    &lt;NA&gt;  9 Samples     &lt;NA&gt;  GPL24676</span></span>
<span><span class="co">#&gt; 2: 200192889             &lt;NA&gt;    &lt;NA&gt;  9 Samples     &lt;NA&gt;  GPL24676</span></span>
<span><span class="co">#&gt; 3: 200266108             &lt;NA&gt;    &lt;NA&gt;  6 Samples     &lt;NA&gt;  GPL24676</span></span>
<span><span class="co">#&gt; 4: 200280402             &lt;NA&gt;    &lt;NA&gt; 16 Samples     &lt;NA&gt;  GPL16791</span></span>
<span><span class="co">#&gt; 5: 200262922             &lt;NA&gt;    &lt;NA&gt;  6 Samples     &lt;NA&gt;  GPL20301</span></span>
<span><span class="co">#&gt;    Series Accession number_of_samples</span></span>
<span><span class="co">#&gt;              &lt;char&gt;             &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1:        GSE255028                 9</span></span>
<span><span class="co">#&gt; 2:        GSE192889                 9</span></span>
<span><span class="co">#&gt; 3:        GSE266108                 6</span></span>
<span><span class="co">#&gt; 4:        GSE280402                16</span></span>
<span><span class="co">#&gt; 5:        GSE262922                 6</span></span></code></pre></div>
<p>After filtering, we got 36 candidate datasets. This can reduce a lot of time of us comparing with refining datasets by reading the summary records.</p>
</div>
<div class="section level3">
<h3 id="download-data-from-geo-database---geo">Download data from GEO database - <code>geo</code>
<a class="anchor" aria-label="anchor" href="#download-data-from-geo-database---geo"></a>
</h3>
<p>GEO database mainly provides SOFT (Simple Omnibus Format in Text) formatted files for GPL, GSM and GDS entity. SOFT is designed for rapid batch submission and download of data. SOFT is a simple line-based, plain text format, meaning that SOFT files may be readily generated from common spreadsheet and database applications. A single SOFT file can hold both data tables and accompanying descriptive information for multiple, concatenated Platforms, Samples, and/or Series records. <code>geokit</code> provide a <code>GEOSoft</code> class object to store SOFT file contents, <code>GEOSoft</code> object contains four slots (“accession”, “meta”, “datatable”, and “columns”). <code>accession</code> slot stores the GEO accession ID, <code>meta</code> slot contains the metadata header in the SOFT formatted file, and <code>datatable</code> slot contains the the data table in SOFT file which is the main data for us to use, along with a <code>columns</code> slot providing descriptive column header for the <code>datatable</code> data. We can use the function with the same name of these slots to extract the data.</p>
<p><code>geo</code> can download SOFT files and preprocess them well, here is some example code to get soft file from <code>GPL</code>, <code>GSM</code> and <code>GDS</code> entity respectively.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gpl</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span><span class="st">"gpl98"</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GPL full amount file from GEO Accession Site</span></span>
<span><span class="va">gpl</span></span>
<span><span class="co">#&gt; An object of GEOSoft</span></span>
<span><span class="co">#&gt; datatable: a 8934 * 16 data.frame</span></span>
<span><span class="co">#&gt; columns: a 16 * 1 data.frame</span></span>
<span><span class="co">#&gt;   columnsData: labelDescription</span></span>
<span><span class="co">#&gt; meta: Platform_contact_address Platform_contact_city</span></span>
<span><span class="co">#&gt;   Platform_contact_country ... Platform_title Platform_web_link (26</span></span>
<span><span class="co">#&gt;   total)</span></span>
<span><span class="co">#&gt; accession: GPL98</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">datatable</a></span><span class="op">(</span><span class="va">gpl</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                          ID   GB_ACC SPOT_ID Species Scientific Name</span></span>
<span><span class="co">#&gt; AA000993_at     AA000993_at AA000993                    Homo sapiens</span></span>
<span><span class="co">#&gt; AA001296_s_at AA001296_s_at AA001296                    Homo sapiens</span></span>
<span><span class="co">#&gt; AA002245_at     AA002245_at AA002245                    Homo sapiens</span></span>
<span><span class="co">#&gt; AA004231_at     AA004231_at AA004231                    Homo sapiens</span></span>
<span><span class="co">#&gt; AA004333_at     AA004333_at AA004333                    Homo sapiens</span></span>
<span><span class="co">#&gt; AA004987_at     AA004987_at AA004987                    Homo sapiens</span></span>
<span><span class="co">#&gt;               Annotation Date     Sequence Type Sequence Source</span></span>
<span><span class="co">#&gt; AA000993_at      Mar 11, 2009 Exemplar sequence         GenBank</span></span>
<span><span class="co">#&gt; AA001296_s_at    Mar 11, 2009 Exemplar sequence         GenBank</span></span>
<span><span class="co">#&gt; AA002245_at      Mar 11, 2009 Exemplar sequence         GenBank</span></span>
<span><span class="co">#&gt; AA004231_at      Mar 11, 2009 Exemplar sequence         GenBank</span></span>
<span><span class="co">#&gt; AA004333_at      Mar 11, 2009 Exemplar sequence         GenBank</span></span>
<span><span class="co">#&gt; AA004987_at      Mar 11, 2009 Exemplar sequence         GenBank</span></span>
<span><span class="co">#&gt;                                                                             Target Description</span></span>
<span><span class="co">#&gt; AA000993_at                 ze46h10.r1 Soares retina N2b4HR Homo sapiens cDNA clone 362083 5'.</span></span>
<span><span class="co">#&gt; AA001296_s_at zh82b09.r1 Soares fetal liver spleen 1NFLS S1 Homo sapiens cDNA clone 427769 5'.</span></span>
<span><span class="co">#&gt; AA002245_at   zh85f01.r1 Soares fetal liver spleen 1NFLS S1 Homo sapiens cDNA clone 428089 5'.</span></span>
<span><span class="co">#&gt; AA004231_at   zh92a03.r1 Soares fetal liver spleen 1NFLS S1 Homo sapiens cDNA clone 428716 5'.</span></span>
<span><span class="co">#&gt; AA004333_at   zh91a01.r1 Soares fetal liver spleen 1NFLS S1 Homo sapiens cDNA clone 428616 5'.</span></span>
<span><span class="co">#&gt; AA004987_at   zh94b01.r1 Soares fetal liver spleen 1NFLS S1 Homo sapiens cDNA clone 428905 5'.</span></span>
<span><span class="co">#&gt;               Representative Public ID</span></span>
<span><span class="co">#&gt; AA000993_at                   AA000993</span></span>
<span><span class="co">#&gt; AA001296_s_at                 AA001296</span></span>
<span><span class="co">#&gt; AA002245_at                   AA002245</span></span>
<span><span class="co">#&gt; AA004231_at                   AA004231</span></span>
<span><span class="co">#&gt; AA004333_at                   AA004333</span></span>
<span><span class="co">#&gt; AA004987_at                   AA004987</span></span>
<span><span class="co">#&gt;                                                                     Gene Title</span></span>
<span><span class="co">#&gt; AA000993_at                                             PR domain containing 8</span></span>
<span><span class="co">#&gt; AA001296_s_at                                            PHD finger protein 23</span></span>
<span><span class="co">#&gt; AA002245_at                               zinc finger CCCH-type containing 11A</span></span>
<span><span class="co">#&gt; AA004231_at                                                                   </span></span>
<span><span class="co">#&gt; AA004333_at                                                                   </span></span>
<span><span class="co">#&gt; AA004987_at   BMP2 inducible kinase, mRNA (cDNA clone MGC:33000 IMAGE:5272264)</span></span>
<span><span class="co">#&gt;               Gene Symbol ENTREZ_GENE_ID       RefSeq Transcript ID</span></span>
<span><span class="co">#&gt; AA000993_at         PRDM8          56978 NM_001099403 /// NM_020226</span></span>
<span><span class="co">#&gt; AA001296_s_at       PHF23          79142                  NM_024297</span></span>
<span><span class="co">#&gt; AA002245_at       ZC3H11A           9877                  NM_014827</span></span>
<span><span class="co">#&gt; AA004231_at                                                        </span></span>
<span><span class="co">#&gt; AA004333_at                                                        </span></span>
<span><span class="co">#&gt; AA004987_at         BMP2K          55589    NM_017593 /// NM_198892</span></span>
<span><span class="co">#&gt;                                                                                                                                               Gene Ontology Biological Process</span></span>
<span><span class="co">#&gt; AA000993_at   0006350 // transcription // inferred from electronic annotation /// 0006355 // regulation of transcription, DNA-dependent // inferred from electronic annotation</span></span>
<span><span class="co">#&gt; AA001296_s_at                                                                                                                                                                 </span></span>
<span><span class="co">#&gt; AA002245_at                                                                                                                                                                   </span></span>
<span><span class="co">#&gt; AA004231_at                                                                                                                                                                   </span></span>
<span><span class="co">#&gt; AA004333_at                                                                                                                                                                   </span></span>
<span><span class="co">#&gt; AA004987_at                                                                               0006468 // protein amino acid phosphorylation // inferred from electronic annotation</span></span>
<span><span class="co">#&gt;                                                                                                            Gene Ontology Cellular Component</span></span>
<span><span class="co">#&gt; AA000993_at   0005622 // intracellular // inferred from electronic annotation /// 0005634 // nucleus // inferred from electronic annotation</span></span>
<span><span class="co">#&gt; AA001296_s_at                                                                                                                              </span></span>
<span><span class="co">#&gt; AA002245_at                                                                                                                                </span></span>
<span><span class="co">#&gt; AA004231_at                                                                                                                                </span></span>
<span><span class="co">#&gt; AA004333_at                                                                                                                                </span></span>
<span><span class="co">#&gt; AA004987_at                                                                       0005634 // nucleus // inferred from electronic annotation</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                   Gene Ontology Molecular Function</span></span>
<span><span class="co">#&gt; AA000993_at                                                                                                                                                                                                                                                           0003677 // DNA binding // inferred from electronic annotation /// 0008270 // zinc ion binding // inferred from electronic annotation /// 0046872 // metal ion binding // inferred from electronic annotation</span></span>
<span><span class="co">#&gt; AA001296_s_at                                                                                                                                                                                                                                                     0005515 // protein binding // inferred from electronic annotation /// 0008270 // zinc ion binding // inferred from electronic annotation /// 0046872 // metal ion binding // inferred from electronic annotation</span></span>
<span><span class="co">#&gt; AA002245_at                                                                                                                                                                                                                                                  0003676 // nucleic acid binding // inferred from electronic annotation /// 0008270 // zinc ion binding // inferred from electronic annotation /// 0046872 // metal ion binding // inferred from electronic annotation</span></span>
<span><span class="co">#&gt; AA004231_at                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </span></span>
<span><span class="co">#&gt; AA004333_at                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </span></span>
<span><span class="co">#&gt; AA004987_at   0000166 // nucleotide binding // inferred from electronic annotation /// 0004672 // protein kinase activity // inferred from electronic annotation /// 0004674 // protein serine/threonine kinase activity // inferred from electronic annotation /// 0005524 // ATP binding // inferred from electronic annotation /// 0016301 // kinase activity // inferred from electronic annotation /// 0016740 // transferase activity // inferred from electronic annotation</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">columns</a></span><span class="op">(</span><span class="va">gpl</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                    labelDescription</span></span>
<span><span class="co">#&gt; ID                                                                                                                                                                                                                                                                   Affymetrix Probe Set ID LINK_PRE:"https://www.affymetrix.com/LinkServlet?array=U35K&amp;probeset="</span></span>
<span><span class="co">#&gt; GB_ACC                                                                                                                                                                                                                                                                               GenBank Accession Number LINK_PRE:"http://www.ncbi.nlm.nih.gov/nuccore/?term="</span></span>
<span><span class="co">#&gt; SPOT_ID                                                                                                                                                                                                                                                                                                                                                        &lt;NA&gt;</span></span>
<span><span class="co">#&gt; Species Scientific Name                                                                                                                                                                                                                                                                         The genus and species of the organism represented by the probe set.</span></span>
<span><span class="co">#&gt; Annotation Date                                                                                                                                                                       The date that the annotations for this probe array were last updated. It will generally be earlier than the date when the annotations were posted on the Affymetrix web site.</span></span>
<span><span class="co">#&gt; Sequence Type           Indicates whether the sequence is an Exemplar, Consensus or Control sequence. An Exemplar is a single nucleotide sequence taken directly from a public database. This sequence could be an mRNA or EST. A Consensus sequence, is a nucleotide sequence assembled by Affymetrix, based on one or more sequence taken from a public database.</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsm</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span><span class="st">"GSM1"</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GSM full amount file from GEO Accession Site</span></span>
<span><span class="va">gsm</span></span>
<span><span class="co">#&gt; An object of GEOSoft</span></span>
<span><span class="co">#&gt; datatable: a 5494 * 3 data.frame</span></span>
<span><span class="co">#&gt; columns: a 3 * 1 data.frame</span></span>
<span><span class="co">#&gt;   columnsData: labelDescription</span></span>
<span><span class="co">#&gt; meta: Sample_anchor Sample_channel_count Sample_contact_address ...</span></span>
<span><span class="co">#&gt;   Sample_title Sample_type (29 total)</span></span>
<span><span class="co">#&gt; accession: GSM1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">datatable</a></span><span class="op">(</span><span class="va">gsm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   TAG COUNT     TPM</span></span>
<span><span class="co">#&gt; AAAAAAAAAA AAAAAAAAAA    17 1741.98</span></span>
<span><span class="co">#&gt; AAAAAAATCA AAAAAAATCA     1  102.47</span></span>
<span><span class="co">#&gt; AAAAAAATTT AAAAAAATTT     1  102.47</span></span>
<span><span class="co">#&gt; AAAAAACAAA AAAAAACAAA     1  102.47</span></span>
<span><span class="co">#&gt; AAAAAACTCC AAAAAACTCC     1  102.47</span></span>
<span><span class="co">#&gt; AAAAAATAAA AAAAAATAAA     1  102.47</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">columns</a></span><span class="op">(</span><span class="va">gsm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;         labelDescription</span></span>
<span><span class="co">#&gt; TAG   Ten base SAGE tag,</span></span>
<span><span class="co">#&gt; COUNT         TAG NUMBER</span></span>
<span><span class="co">#&gt; TPM     tags per million</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gds</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span><span class="st">"GDS10"</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GDS soft file from FTP site</span></span>
<span><span class="va">gds</span></span>
<span><span class="co">#&gt; An object of GEOSoft</span></span>
<span><span class="co">#&gt; datatable: a 39114 * 30 data.frame</span></span>
<span><span class="co">#&gt; columns: a 30 * 4 data.frame</span></span>
<span><span class="co">#&gt;   columnsData: labelDescription subset_dataset_id subset_description</span></span>
<span><span class="co">#&gt;     subset_type</span></span>
<span><span class="co">#&gt; meta: Database_email Database_institute Database_name ...</span></span>
<span><span class="co">#&gt;   dataset_update_date dataset_value_type (21 total)</span></span>
<span><span class="co">#&gt; accession: GDS10</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">datatable</a></span><span class="op">(</span><span class="va">gds</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;   ID_REF    IDENTIFIER GSM582 GSM589 GSM583 GSM590 GSM584 GSM591 GSM585 GSM592</span></span>
<span><span class="co">#&gt; 1      1 1200011I18Rik    101     54    111     55     87     30     99     43</span></span>
<span><span class="co">#&gt; 2      2             2     26     23     30     27     19     22     32     19</span></span>
<span><span class="co">#&gt; 3      3       Ccdc28b     NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 4      4      AA014405    233    162    252    178    214    144    238    147</span></span>
<span><span class="co">#&gt; 5      5        Crebrf     NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 6      6             6    691    661    696    652    609    665    684    672</span></span>
<span><span class="co">#&gt;   GSM586 GSM593 GSM587 GSM594 GSM588 GSM595 GSM596 GSM603 GSM597 GSM604 GSM598</span></span>
<span><span class="co">#&gt; 1    105     56     43     14    112     43     97     36    117     40    125</span></span>
<span><span class="co">#&gt; 2     24     25     14     49     32     29     31     22     26     26     35</span></span>
<span><span class="co">#&gt; 3     NA     NA     NA      7     NA      4     10     22     NA     15     NA</span></span>
<span><span class="co">#&gt; 4    250    166     86     22    236    139    216    112    241    130    270</span></span>
<span><span class="co">#&gt; 5     NA     NA     NA     NA     NA      3     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 6    644    679    631    596    609    606    601    557    596    580    601</span></span>
<span><span class="co">#&gt;   GSM605 GSM599 GSM606 GSM600 GSM607 GSM601 GSM608 GSM602 GSM609</span></span>
<span><span class="co">#&gt; 1     45     99      1    109     38     87     18     72     16</span></span>
<span><span class="co">#&gt; 2     26     18     13     25     32     28     40     14     41</span></span>
<span><span class="co">#&gt; 3     23     NA     29      9     25     11     40     NA     22</span></span>
<span><span class="co">#&gt; 4    144    239    148    211    139    208     16    174     15</span></span>
<span><span class="co">#&gt; 5     NA     NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 6    554    562    561    580    568    519    562    497    564</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">columns</a></span><span class="op">(</span><span class="va">gds</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                  labelDescription   subset_dataset_id</span></span>
<span><span class="co">#&gt; ID_REF              Platform reference identifier                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; IDENTIFIER                             identifier                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; GSM582      Value for GSM582: NOD_S1; src: Spleen GDS10; GDS10; GDS10</span></span>
<span><span class="co">#&gt; GSM589      Value for GSM589: NOD_S2; src: Spleen GDS10; GDS10; GDS10</span></span>
<span><span class="co">#&gt; GSM583     Value for GSM583: Idd3_S1; src: Spleen GDS10; GDS10; GDS10</span></span>
<span><span class="co">#&gt; GSM590     Value for GSM590: Idd3_S2; src: Spleen GDS10; GDS10; GDS10</span></span>
<span><span class="co">#&gt;                          subset_description                   subset_type</span></span>
<span><span class="co">#&gt; ID_REF                                 &lt;NA&gt;                          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; IDENTIFIER                             &lt;NA&gt;                          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; GSM582                spleen; NOD; diabetic tissue; strain; disease state</span></span>
<span><span class="co">#&gt; GSM589                spleen; NOD; diabetic tissue; strain; disease state</span></span>
<span><span class="co">#&gt; GSM583     spleen; Idd3; diabetic-resistant tissue; strain; disease state</span></span>
<span><span class="co">#&gt; GSM590     spleen; Idd3; diabetic-resistant tissue; strain; disease state</span></span></code></pre></div>
<p>For GSE entity, there is also a soft file associated with it. But the structure is different with <code>GPL</code>, <code>GSM</code> and <code>GDS</code> entity, <code>geokit</code> provide <code>GEOSeries</code> class to keep contents in GSE soft file. Actually, a GSE soft file contains almost all contents in its subsets soft file including both <code>GPL</code> and <code>GSM</code>, so <code>GEOSeries</code> class provides both <code>gpl</code> and <code>gsm</code> slots as a list of <code>GEOSoft</code>. To download GSE soft file, we just set <code>gse_matrix</code> to <code>FALSE</code> in <code>geo</code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span><span class="st">"GSE10"</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, gse_matrix <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GSE soft file from FTP site</span></span>
<span><span class="va">gse</span></span>
<span><span class="co">#&gt; An object of GEOSeries</span></span>
<span><span class="co">#&gt; gsm: GSM571 GSM572 GSM573 GSM574</span></span>
<span><span class="co">#&gt; gpl: GPL4</span></span>
<span><span class="co">#&gt; meta: Database_email Database_institute Database_name ... Series_title</span></span>
<span><span class="co">#&gt;   Series_type (31 total)</span></span>
<span><span class="co">#&gt; accession: GSE10</span></span></code></pre></div>
<p>It’s more common to use a series matrix file in our usual analysis workflow, we can also handle it easily in <code>geokit</code>, as what we need to do is just set <code>gse_matrix</code> to <code>TRUE</code> in <code>geo</code> function, which is also the default value. When <code>gse_matrix</code> is <code>TRUE</code>, <code>geo</code> will return a <code>ExpressionSet</code> object which can interact with lots of Bioconductor packages. There are two parameters controling the processing details when parsing series matrix file. When parsing phenoData from series matrix files directly, it’s common to fail to discern <code>characteristics_ch*</code> columns, which contain the important traits informations of corresponding samples, since many <code>characteristics_ch*</code> columns in series matrix files often lacks separate strings. <code>pdata_from_soft</code>, which indicates whether retrieve phenoData from GEO Soft file, can help handle this problem well. When the soft file is large and we don’t want to use it, we can set <code>pdata_from_soft</code> to <code>FALSE</code> and use <code>parse_pdata</code> function to parse it manully. Another important parameter is <code>add_gpl</code>, where <code>FALSE</code> indicates <code>geo</code> will try to map the current GPL accession id into a Bioconductor annotation package, then we can use the latest bioconductor annotation package to get the up-to-date featureData, otherwise, <code>geo</code> will add featureData from GPL soft file directly.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse_matix</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span><span class="st">"GSE10"</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GSE matrix file from FTP site</span></span>
<span><span class="co">#&gt; Finding 1 {.strong GSE} {.field soft} file already downloaded:</span></span>
<span><span class="co">#&gt; 'GSE10_family.soft.gz'</span></span>
<span><span class="co">#&gt; → Parsing series soft file 'GSE10_family.soft.gz'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ✔ Parsing 1 series soft file successfully!</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; → Parsing 1 series matrix file of GSE10</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ✔ Parsing 1 GSE series matrix successfully!</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; → Constructing &lt;ExpressionSet&gt;</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ✔ Found Bioconductor annotation package for "GPL4"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Downloading 1 GPL annot file from FTP site</span></span>
<span><span class="co">#&gt; ℹ annot file in FTP site for "GPL4" is not available, so will use data amount file from GEO Accession Site instead</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Downloading 1 GPL data amount file from GEO Accession Site</span></span>
<span><span class="va">gse_matix</span></span>
<span><span class="co">#&gt; ExpressionSet (storageMode: lockedEnvironment)</span></span>
<span><span class="co">#&gt; assayData: 96903 features, 4 samples </span></span>
<span><span class="co">#&gt;   element names: exprs </span></span>
<span><span class="co">#&gt; protocolData: none</span></span>
<span><span class="co">#&gt; phenoData</span></span>
<span><span class="co">#&gt;   sampleNames: GSM571 GSM572 GSM573 GSM574</span></span>
<span><span class="co">#&gt;   varLabels: anchor channel_count ... type (32 total)</span></span>
<span><span class="co">#&gt;   varMetadata: labelDescription</span></span>
<span><span class="co">#&gt; featureData</span></span>
<span><span class="co">#&gt;   featureNames: AAAAAAAAAA AAAAAAAAAC ... TTTTTTTTTT (96903 total)</span></span>
<span><span class="co">#&gt;   fvarLabels: TAG GI</span></span>
<span><span class="co">#&gt;   fvarMetadata: labelDescription</span></span>
<span><span class="co">#&gt; experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">#&gt;   pubMedIds: 11756676 </span></span>
<span><span class="co">#&gt; Annotation: GPL4</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse_matrix_with_pdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span></span>
<span>    <span class="st">"gse53987"</span>,</span>
<span>    odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    pdata_from_soft <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    add_gpl <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GSE matrix file from FTP site</span></span>
<span><span class="co">#&gt; → Parsing 1 series matrix file of GSE53987</span></span>
<span><span class="co">#&gt; Warning: Cannot parse characteristic column correctly</span></span>
<span><span class="co">#&gt; ℹ Details see "characteristics_ch1" column in phenoData</span></span>
<span><span class="co">#&gt; ℹ Please use `parse_pdata()` or `parse_gsm_list()` function to convert it</span></span>
<span><span class="co">#&gt;   manually if necessary!</span></span>
<span><span class="co">#&gt; ✔ Parsing 1 GSE series matrix successfully!</span></span>
<span><span class="co">#&gt; → Constructing &lt;ExpressionSet&gt;</span></span>
<span><span class="co">#&gt; ✔ Found Bioconductor annotation package for "GPL570"</span></span>
<span><span class="va">gse_matrix_smp_info</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse_matrix_with_pdata</span><span class="op">)</span></span>
<span><span class="va">gse_matrix_smp_info</span><span class="op">$</span><span class="va">characteristics_ch1</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span></span>
<span>    <span class="va">gse_matrix_smp_info</span><span class="op">$</span><span class="va">characteristics_ch1</span>,</span>
<span>    <span class="st">"gender|race|pmi|ph|rin|tissue|disease state"</span>,</span>
<span>    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"; "</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">gse_matrix_smp_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/parse_pdata.html">parse_pdata</a></span><span class="op">(</span><span class="va">gse_matrix_smp_info</span><span class="op">)</span></span>
<span><span class="va">gse_matrix_smp_info</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span></span>
<span>    <span class="st">"^ch1_|characteristics_ch1"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gse_matrix_smp_info</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt;            ch1_age ch1_gender ch1_race ch1_pmi ch1_ph ch1_rin</span></span>
<span><span class="co">#&gt; GSM1304852      52          M        W   23.50   6.70     6.3</span></span>
<span><span class="co">#&gt; GSM1304853      50          F        W   11.70   6.40     6.8</span></span>
<span><span class="co">#&gt; GSM1304854      28          F        W   22.30   6.30     7.7</span></span>
<span><span class="co">#&gt; GSM1304855      55          F        W   17.50   6.40     7.6</span></span>
<span><span class="co">#&gt; GSM1304856      58          M        W   27.70   6.80     7.0</span></span>
<span><span class="co">#&gt; GSM1304857      28          M        W   27.40   6.20     7.7</span></span>
<span><span class="co">#&gt; GSM1304858      49          F        W   21.50   6.70     8.2</span></span>
<span><span class="co">#&gt; GSM1304859      42          F        W   31.20   6.50     5.6</span></span>
<span><span class="co">#&gt; GSM1304860      43          F        W   31.90   6.70     6.3</span></span>
<span><span class="co">#&gt; GSM1304861      50          M        W   12.10   6.70     7.4</span></span>
<span><span class="co">#&gt; GSM1304862      40          M        W   18.50   6.40     6.5</span></span>
<span><span class="co">#&gt; GSM1304863      39          F        W   22.20   6.70     7.9</span></span>
<span><span class="co">#&gt; GSM1304864      45          M        W   27.20   7.10     8.1</span></span>
<span><span class="co">#&gt; GSM1304865      42          M        W   12.50   6.70     8.2</span></span>
<span><span class="co">#&gt; GSM1304866      65          M        W    8.90   6.70     6.6</span></span>
<span><span class="co">#&gt; GSM1304867      51          F        W   21.50   6.70     7.0</span></span>
<span><span class="co">#&gt; GSM1304868      39          M        W   24.20   6.60     7.8</span></span>
<span><span class="co">#&gt; GSM1304869      48          M        W   18.10   6.90     7.0</span></span>
<span><span class="co">#&gt; GSM1304870      51          M        W   24.20   6.60     7.8</span></span>
<span><span class="co">#&gt; GSM1304871      51          F        W    7.80   6.60     7.2</span></span>
<span><span class="co">#&gt; GSM1304872      36          F        W   14.50   6.40     8.0</span></span>
<span><span class="co">#&gt; GSM1304873      65          F        W   18.50   6.50     7.0</span></span>
<span><span class="co">#&gt; GSM1304874      55          M        W   28.00   6.10     6.8</span></span>
<span><span class="co">#&gt; GSM1304875      22          M        W   20.10   6.80     7.1</span></span>
<span><span class="co">#&gt; GSM1304876      52          F        W   22.60   7.10     7.0</span></span>
<span><span class="co">#&gt; GSM1304877      58          F        W   22.70   6.40     6.3</span></span>
<span><span class="co">#&gt; GSM1304878      40          F        B   16.60   6.80     7.9</span></span>
<span><span class="co">#&gt; GSM1304879      41          F        W   15.40   6.60     8.5</span></span>
<span><span class="co">#&gt; GSM1304880      49          M        W   21.20   6.50     7.8</span></span>
<span><span class="co">#&gt; GSM1304881      48          M        W   21.68   6.60     7.3</span></span>
<span><span class="co">#&gt; GSM1304882      39          F        W   24.50   6.80     8.2</span></span>
<span><span class="co">#&gt; GSM1304883      48          M        W   24.50   6.50     7.0</span></span>
<span><span class="co">#&gt; GSM1304884      43          M        W   13.80   6.60     7.6</span></span>
<span><span class="co">#&gt; GSM1304885      68          M        W   11.80   6.80     6.1</span></span>
<span><span class="co">#&gt; GSM1304886      58          F        W   18.80   6.60     7.2</span></span>
<span><span class="co">#&gt; GSM1304887      43          M        W   22.30   6.70     7.9</span></span>
<span><span class="co">#&gt; GSM1304888      51          M        W   24.60   6.50     7.7</span></span>
<span><span class="co">#&gt; GSM1304889      53          F        W   11.90   6.70     8.1</span></span>
<span><span class="co">#&gt; GSM1304890      26          F        W   13.40   6.40     7.5</span></span>
<span><span class="co">#&gt; GSM1304891      52          F        W   10.30   6.50     6.6</span></span>
<span><span class="co">#&gt; GSM1304892      62          M        W   26.00   6.50     6.8</span></span>
<span><span class="co">#&gt; GSM1304893      29          M        W   26.60   6.90     7.8</span></span>
<span><span class="co">#&gt; GSM1304894      49          F        W   23.40   6.40     6.2</span></span>
<span><span class="co">#&gt; GSM1304895      54          F        W   17.90   6.20     6.1</span></span>
<span><span class="co">#&gt; GSM1304896      28          F        B   24.80   6.60     8.2</span></span>
<span><span class="co">#&gt; GSM1304897      42          M        W   14.30   6.40     6.2</span></span>
<span><span class="co">#&gt; GSM1304898      44          M        W   19.30   6.50     6.3</span></span>
<span><span class="co">#&gt; GSM1304899      40          F        W   22.20   6.60     8.0</span></span>
<span><span class="co">#&gt; GSM1304900      47          M        W   24.00   6.60     5.5</span></span>
<span><span class="co">#&gt; GSM1304901      59          M        W   13.00   6.60     7.2</span></span>
<span><span class="co">#&gt; GSM1304902      47          F        W   22.30   6.60     6.5</span></span>
<span><span class="co">#&gt; GSM1304903      34          M        W   24.40   6.60     8.4</span></span>
<span><span class="co">#&gt; GSM1304904      51          M        W   28.30   7.30     7.0</span></span>
<span><span class="co">#&gt; GSM1304905      49          M        W   21.50   5.97     6.0</span></span>
<span><span class="co">#&gt; GSM1304906      47          F        W   14.37   6.35     6.3</span></span>
<span><span class="co">#&gt; GSM1304907      25          F        B   20.10   6.73     5.6</span></span>
<span><span class="co">#&gt; GSM1304908      62          M        W   22.70   7.14     6.3</span></span>
<span><span class="co">#&gt; GSM1304909      44          F        W   24.50   6.63     7.8</span></span>
<span><span class="co">#&gt; GSM1304910      46          F        W   23.80   6.61     6.9</span></span>
<span><span class="co">#&gt; GSM1304911      50          M        W   11.00   6.23     7.2</span></span>
<span><span class="co">#&gt; GSM1304912      46          M        W   15.80   6.19     6.2</span></span>
<span><span class="co">#&gt; GSM1304913      41          F        W   20.10   6.27     6.7</span></span>
<span><span class="co">#&gt; GSM1304914      47          M        W   28.90   6.58     6.7</span></span>
<span><span class="co">#&gt; GSM1304915      37          M        B    5.98   6.07     6.4</span></span>
<span><span class="co">#&gt; GSM1304916      58          M        W    7.70   6.22     6.7</span></span>
<span><span class="co">#&gt; GSM1304917      44          F        B   18.70   6.20     6.4</span></span>
<span><span class="co">#&gt; GSM1304918      38          M        W   28.80   6.56     6.6</span></span>
<span><span class="co">#&gt; GSM1304919      52          M        B   27.10   6.68     6.3</span></span>
<span><span class="co">#&gt; GSM1304920      52          M        W   23.50   6.70     7.2</span></span>
<span><span class="co">#&gt; GSM1304921      50          F        W   11.70   6.40     8.6</span></span>
<span><span class="co">#&gt; GSM1304922      28          F        W   22.30   6.30     8.6</span></span>
<span><span class="co">#&gt; GSM1304923      55          F        W   17.50   6.40     8.0</span></span>
<span><span class="co">#&gt; GSM1304924      58          M        W   27.70   6.80     7.5</span></span>
<span><span class="co">#&gt; GSM1304925      28          M        W   27.40   6.20     7.9</span></span>
<span><span class="co">#&gt; GSM1304926      49          F        W   21.50   6.70     8.1</span></span>
<span><span class="co">#&gt; GSM1304927      56          F        W   24.50   6.10     6.9</span></span>
<span><span class="co">#&gt; GSM1304928      50          M        W   12.10   6.70     7.6</span></span>
<span><span class="co">#&gt; GSM1304929      40          M        W   18.50   6.40     7.9</span></span>
<span><span class="co">#&gt; GSM1304930      39          F        W   22.20   6.70     7.8</span></span>
<span><span class="co">#&gt; GSM1304931      45          M        W   27.20   7.10     7.3</span></span>
<span><span class="co">#&gt; GSM1304932      42          M        W   12.50   6.70     7.6</span></span>
<span><span class="co">#&gt; GSM1304933      65          M        W    8.90   6.70     6.9</span></span>
<span><span class="co">#&gt; GSM1304934      51          F        W   21.50   6.70     7.7</span></span>
<span><span class="co">#&gt; GSM1304935      39          M        W   24.20   6.60     7.3</span></span>
<span><span class="co">#&gt; GSM1304936      48          M        W   18.10   6.90     8.2</span></span>
<span><span class="co">#&gt; GSM1304937      51          M        W   24.20   6.60     7.9</span></span>
<span><span class="co">#&gt; GSM1304938      51          F        W    7.80   6.60     8.6</span></span>
<span><span class="co">#&gt; GSM1304939      36          F        W   14.50   6.40     8.6</span></span>
<span><span class="co">#&gt; GSM1304940      65          F        W   18.50   6.50     8.3</span></span>
<span><span class="co">#&gt; GSM1304941      55          M        W   28.00   6.10     7.9</span></span>
<span><span class="co">#&gt; GSM1304942      22          M        W   20.10   6.80     8.1</span></span>
<span><span class="co">#&gt; GSM1304943      52          F        W   22.60   7.10     8.2</span></span>
<span><span class="co">#&gt; GSM1304944      58          F        W   22.70   6.40     8.0</span></span>
<span><span class="co">#&gt; GSM1304945      40          F        B   16.60   6.80     8.2</span></span>
<span><span class="co">#&gt; GSM1304946      41          F        W   15.40   6.60     8.2</span></span>
<span><span class="co">#&gt; GSM1304947      49          M        W   21.20   6.50     7.9</span></span>
<span><span class="co">#&gt; GSM1304948      48          M        W   21.68   6.60     7.5</span></span>
<span><span class="co">#&gt; GSM1304949      39          F        W   24.50   6.80     7.4</span></span>
<span><span class="co">#&gt; GSM1304950      48          M        W   24.50   6.50     6.8</span></span>
<span><span class="co">#&gt; GSM1304951      43          M        W   13.80   6.60     7.5</span></span>
<span><span class="co">#&gt; GSM1304952      68          M        W   11.80   6.80     6.7</span></span>
<span><span class="co">#&gt; GSM1304953      58          F        W   18.80   6.60     8.7</span></span>
<span><span class="co">#&gt; GSM1304954      43          M        W   22.30   6.70     8.0</span></span>
<span><span class="co">#&gt; GSM1304955      46          M        W   22.00   6.30     6.6</span></span>
<span><span class="co">#&gt; GSM1304956      51          M        W   24.60   6.50     7.8</span></span>
<span><span class="co">#&gt; GSM1304957      53          F        W   11.90   6.70     8.4</span></span>
<span><span class="co">#&gt; GSM1304958      26          F        W   13.40   6.40     8.4</span></span>
<span><span class="co">#&gt; GSM1304959      52          F        W   10.30   6.50     8.1</span></span>
<span><span class="co">#&gt; GSM1304960      62          M        W   26.00   6.50     7.8</span></span>
<span><span class="co">#&gt; GSM1304961      29          M        W   26.60   6.90     8.2</span></span>
<span><span class="co">#&gt; GSM1304962      49          F        W   23.40   6.40     7.6</span></span>
<span><span class="co">#&gt; GSM1304963      54          F        W   17.90   6.20     7.5</span></span>
<span><span class="co">#&gt; GSM1304964      28          F        B   24.80   6.60     7.9</span></span>
<span><span class="co">#&gt; GSM1304965      42          M        W   14.30   6.40     8.4</span></span>
<span><span class="co">#&gt; GSM1304966      40          F        W   22.20   6.60     7.7</span></span>
<span><span class="co">#&gt; GSM1304967      47          M        W   24.00   6.60     6.8</span></span>
<span><span class="co">#&gt; GSM1304968      44          M        W   11.00   6.50     7.2</span></span>
<span><span class="co">#&gt; GSM1304969      59          M        W   13.00   6.60     7.7</span></span>
<span><span class="co">#&gt; GSM1304970      47          F        W   22.30   6.60     6.7</span></span>
<span><span class="co">#&gt; GSM1304971      34          M        W   24.40   6.60     7.8</span></span>
<span><span class="co">#&gt; GSM1304972      51          M        W   28.30   7.30     7.7</span></span>
<span><span class="co">#&gt; GSM1304973      49          M        W   21.50   5.97     7.0</span></span>
<span><span class="co">#&gt; GSM1304974      47          F        W   14.37   6.35     9.0</span></span>
<span><span class="co">#&gt; GSM1304975      25          F        B   20.10   6.73     7.2</span></span>
<span><span class="co">#&gt; GSM1304976      41          F        W   17.10   6.90     8.3</span></span>
<span><span class="co">#&gt; GSM1304977      62          M        W   22.70   7.14     8.1</span></span>
<span><span class="co">#&gt; GSM1304978      47          F        B   20.10   7.30     8.1</span></span>
<span><span class="co">#&gt; GSM1304979      44          F        W   24.50   6.63     7.6</span></span>
<span><span class="co">#&gt; GSM1304980      46          F        W   23.80   6.61     8.0</span></span>
<span><span class="co">#&gt; GSM1304981      50          M        W   11.00   6.23     8.4</span></span>
<span><span class="co">#&gt; GSM1304982      41          F        W   20.10   6.27     7.4</span></span>
<span><span class="co">#&gt; GSM1304983      47          M        W   28.90   6.58     7.0</span></span>
<span><span class="co">#&gt; GSM1304984      37          M        B    5.98   6.07     6.3</span></span>
<span><span class="co">#&gt; GSM1304985      58          M        W    7.70   6.22     7.3</span></span>
<span><span class="co">#&gt; GSM1304986      44          F        B   18.70   6.20     7.6</span></span>
<span><span class="co">#&gt; GSM1304987      52          M        B   27.10   6.68     7.4</span></span>
<span><span class="co">#&gt; GSM1304988      50          M        W   12.10   6.70     8.6</span></span>
<span><span class="co">#&gt; GSM1304989      40          M        W   18.50   6.40     8.4</span></span>
<span><span class="co">#&gt; GSM1304990      39          F        W   22.20   6.70     9.1</span></span>
<span><span class="co">#&gt; GSM1304991      45          M        W   27.20   7.10     8.7</span></span>
<span><span class="co">#&gt; GSM1304992      42          M        W   12.50   6.70     8.7</span></span>
<span><span class="co">#&gt; GSM1304993      65          M        W    8.90   6.70     8.3</span></span>
<span><span class="co">#&gt; GSM1304994      51          F        W   21.50   6.70     8.4</span></span>
<span><span class="co">#&gt; GSM1304995      39          M        W   24.20   6.60     8.5</span></span>
<span><span class="co">#&gt; GSM1304996      48          M        W   18.10   6.90     8.8</span></span>
<span><span class="co">#&gt; GSM1304997      52          M        W   23.50   6.70     9.1</span></span>
<span><span class="co">#&gt; GSM1304998      50          F        W   11.70   6.40     8.4</span></span>
<span><span class="co">#&gt; GSM1304999      28          F        W   22.30   6.30     9.0</span></span>
<span><span class="co">#&gt; GSM1305000      55          F        W   17.50   6.40     6.0</span></span>
<span><span class="co">#&gt; GSM1305001      58          M        W   27.70   6.80     6.6</span></span>
<span><span class="co">#&gt; GSM1305002      49          F        W   21.50   6.70     8.7</span></span>
<span><span class="co">#&gt; GSM1305003      56          F        W   24.50   6.10     7.7</span></span>
<span><span class="co">#&gt; GSM1305004      42          F        W   31.20   6.50     6.8</span></span>
<span><span class="co">#&gt; GSM1305005      49          M        W   21.20   6.50     8.4</span></span>
<span><span class="co">#&gt; GSM1305006      48          M        W   21.68   6.60     7.5</span></span>
<span><span class="co">#&gt; GSM1305007      39          F        W   24.50   6.80     7.5</span></span>
<span><span class="co">#&gt; GSM1305008      48          M        W   24.50   6.50     7.6</span></span>
<span><span class="co">#&gt; GSM1305009      43          M        W   13.80   6.60     8.7</span></span>
<span><span class="co">#&gt; GSM1305010      68          M        W   11.80   6.80     8.5</span></span>
<span><span class="co">#&gt; GSM1305011      58          F        W   18.80   6.60     8.6</span></span>
<span><span class="co">#&gt; GSM1305012      43          M        W   22.30   6.70     8.5</span></span>
<span><span class="co">#&gt; GSM1305013      46          M        W   22.00   6.30     7.0</span></span>
<span><span class="co">#&gt; GSM1305014      51          M        W   24.20   6.60     8.3</span></span>
<span><span class="co">#&gt; GSM1305015      51          F        W    7.80   6.60     9.0</span></span>
<span><span class="co">#&gt; GSM1305016      36          F        W   14.50   6.40     9.3</span></span>
<span><span class="co">#&gt; GSM1305017      65          F        W   18.50   6.50     7.4</span></span>
<span><span class="co">#&gt; GSM1305018      55          M        W   28.00   6.10     7.6</span></span>
<span><span class="co">#&gt; GSM1305019      22          M        W   20.10   6.80     7.4</span></span>
<span><span class="co">#&gt; GSM1305020      52          F        W   22.60   7.10     8.8</span></span>
<span><span class="co">#&gt; GSM1305021      58          F        W   22.70   6.40     9.0</span></span>
<span><span class="co">#&gt; GSM1305022      40          F        B   16.60   6.80     8.7</span></span>
<span><span class="co">#&gt; GSM1305023      42          M        W   14.30   6.40     8.7</span></span>
<span><span class="co">#&gt; GSM1305024      44          M        W   19.30   6.50     8.5</span></span>
<span><span class="co">#&gt; GSM1305025      47          M        W   24.00   6.60     7.3</span></span>
<span><span class="co">#&gt; GSM1305026      44          M        W   11.00   6.50     7.7</span></span>
<span><span class="co">#&gt; GSM1305027      59          M        W   13.00   6.60     8.4</span></span>
<span><span class="co">#&gt; GSM1305028      47          F        W   22.30   6.60     8.2</span></span>
<span><span class="co">#&gt; GSM1305029      34          M        W   24.40   6.60     9.1</span></span>
<span><span class="co">#&gt; GSM1305030      51          M        W   28.30   7.30     8.6</span></span>
<span><span class="co">#&gt; GSM1305031      51          M        W   24.60   6.50     8.3</span></span>
<span><span class="co">#&gt; GSM1305032      53          F        W   11.90   6.70     8.8</span></span>
<span><span class="co">#&gt; GSM1305033      26          F        W   13.40   6.40     9.2</span></span>
<span><span class="co">#&gt; GSM1305034      52          F        W   10.30   6.50     6.7</span></span>
<span><span class="co">#&gt; GSM1305035      62          M        W   26.00   6.50     7.5</span></span>
<span><span class="co">#&gt; GSM1305036      29          M        W   26.60   6.90     9.2</span></span>
<span><span class="co">#&gt; GSM1305037      49          F        W   23.40   6.40     6.7</span></span>
<span><span class="co">#&gt; GSM1305038      54          F        W   17.90   6.20     9.0</span></span>
<span><span class="co">#&gt; GSM1305039      50          M        W   11.00   6.23     8.5</span></span>
<span><span class="co">#&gt; GSM1305040      46          M        W   15.80   6.19     7.8</span></span>
<span><span class="co">#&gt; GSM1305041      41          F        W   20.10   6.27     8.6</span></span>
<span><span class="co">#&gt; GSM1305042      47          M        W   28.90   6.58     8.4</span></span>
<span><span class="co">#&gt; GSM1305043      37          M        B    5.98   6.07     6.9</span></span>
<span><span class="co">#&gt; GSM1305044      58          M        W    7.70   6.22     6.7</span></span>
<span><span class="co">#&gt; GSM1305045      44          F        B   18.70   6.20     6.9</span></span>
<span><span class="co">#&gt; GSM1305046      38          M        W   28.80   6.56     6.8</span></span>
<span><span class="co">#&gt; GSM1305047      52          M        B   27.10   6.68     8.5</span></span>
<span><span class="co">#&gt; GSM1305048      49          M        W   21.50   5.97     8.4</span></span>
<span><span class="co">#&gt; GSM1305049      47          F        W   14.37   6.35     8.9</span></span>
<span><span class="co">#&gt; GSM1305050      25          F        B   20.10   6.73     7.3</span></span>
<span><span class="co">#&gt; GSM1305051      41          F        W   17.10   6.90     7.3</span></span>
<span><span class="co">#&gt; GSM1305052      62          M        W   22.70   7.14     7.8</span></span>
<span><span class="co">#&gt; GSM1305053      32          M        W   30.80   6.18     7.1</span></span>
<span><span class="co">#&gt; GSM1305054      47          F        B   20.10   7.30     8.8</span></span>
<span><span class="co">#&gt; GSM1305055      50          F        B   22.90   6.25     8.0</span></span>
<span><span class="co">#&gt; GSM1305056      44          F        W   24.50   6.63     9.0</span></span>
<span><span class="co">#&gt;                           ch1_tissue         ch1_disease state</span></span>
<span><span class="co">#&gt; GSM1304852               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304853               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304854               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304855               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304856               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304857               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304858               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304859               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304860               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304861               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304862               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304863               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304864               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304865               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304866               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304867               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304868               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304869               hippocampus          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304870               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304871               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304872               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304873               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304874               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304875               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304876               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304877               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304878               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304879               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304880               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304881               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304882               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304883               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304884               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304885               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304886               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304887               hippocampus                   control</span></span>
<span><span class="co">#&gt; GSM1304888               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304889               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304890               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304891               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304892               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304893               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304894               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304895               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304896               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304897               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304898               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304899               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304900               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304901               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304902               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304903               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304904               hippocampus major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304905               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304906               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304907               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304908               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304909               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304910               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304911               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304912               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304913               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304914               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304915               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304916               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304917               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304918               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304919               hippocampus                    schizo</span></span>
<span><span class="co">#&gt; GSM1304920 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304921 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304922 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304923 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304924 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304925 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304926 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304927 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304928 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304929 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304930 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304931 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304932 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304933 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304934 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304935 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304936 Pre-frontal cortex (BA46)          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304937 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304938 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304939 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304940 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304941 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304942 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304943 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304944 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304945 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304946 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304947 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304948 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304949 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304950 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304951 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304952 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304953 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304954 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304955 Pre-frontal cortex (BA46)                   control</span></span>
<span><span class="co">#&gt; GSM1304956 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304957 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304958 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304959 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304960 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304961 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304962 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304963 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304964 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304965 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304966 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304967 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304968 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304969 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304970 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304971 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304972 Pre-frontal cortex (BA46) major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304973 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304974 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304975 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304976 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304977 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304978 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304979 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304980 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304981 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304982 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304983 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304984 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304985 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304986 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304987 Pre-frontal cortex (BA46)                    schizo</span></span>
<span><span class="co">#&gt; GSM1304988      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304989      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304990      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304991      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304992      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304993      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304994      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304995      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304996      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304997      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304998      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304999      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305000      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305001      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305002      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305003      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305004      Associative striatum          bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305005      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305006      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305007      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305008      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305009      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305010      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305011      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305012      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305013      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305014      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305015      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305016      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305017      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305018      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305019      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305020      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305021      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305022      Associative striatum                   control</span></span>
<span><span class="co">#&gt; GSM1305023      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305024      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305025      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305026      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305027      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305028      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305029      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305030      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305031      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305032      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305033      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305034      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305035      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305036      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305037      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305038      Associative striatum major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305039      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305040      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305041      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305042      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305043      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305044      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305045      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305046      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305047      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305048      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305049      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305050      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305051      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305052      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305053      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305054      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305055      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt; GSM1305056      Associative striatum                    schizo</span></span>
<span><span class="co">#&gt;                                                                                                                               characteristics_ch1</span></span>
<span><span class="co">#&gt; GSM1304852                        age: 52; gender: M; race: W; pmi: 23.5; ph: 6.7; rin: 6.3; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304853                        age: 50; gender: F; race: W; pmi: 11.7; ph: 6.4; rin: 6.8; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304854                        age: 28; gender: F; race: W; pmi: 22.3; ph: 6.3; rin: 7.7; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304855                        age: 55; gender: F; race: W; pmi: 17.5; ph: 6.4; rin: 7.6; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304856                          age: 58; gender: M; race: W; pmi: 27.7; ph: 6.8; rin: 7; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304857                        age: 28; gender: M; race: W; pmi: 27.4; ph: 6.2; rin: 7.7; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304858                        age: 49; gender: F; race: W; pmi: 21.5; ph: 6.7; rin: 8.2; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304859                        age: 42; gender: F; race: W; pmi: 31.2; ph: 6.5; rin: 5.6; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304860                        age: 43; gender: F; race: W; pmi: 31.9; ph: 6.7; rin: 6.3; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304861                        age: 50; gender: M; race: W; pmi: 12.1; ph: 6.7; rin: 7.4; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304862                        age: 40; gender: M; race: W; pmi: 18.5; ph: 6.4; rin: 6.5; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304863                        age: 39; gender: F; race: W; pmi: 22.2; ph: 6.7; rin: 7.9; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304864                        age: 45; gender: M; race: W; pmi: 27.2; ph: 7.1; rin: 8.1; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304865                        age: 42; gender: M; race: W; pmi: 12.5; ph: 6.7; rin: 8.2; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304866                         age: 65; gender: M; race: W; pmi: 8.9; ph: 6.7; rin: 6.6; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304867                          age: 51; gender: F; race: W; pmi: 21.5; ph: 6.7; rin: 7; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304868                        age: 39; gender: M; race: W; pmi: 24.2; ph: 6.6; rin: 7.8; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304869                          age: 48; gender: M; race: W; pmi: 18.1; ph: 6.9; rin: 7; tissue: hippocampus; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304870                                 age: 51; gender: M; race: W; pmi: 24.2; ph: 6.6; rin: 7.8; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304871                                  age: 51; gender: F; race: W; pmi: 7.8; ph: 6.6; rin: 7.2; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304872                                   age: 36; gender: F; race: W; pmi: 14.5; ph: 6.4; rin: 8; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304873                                   age: 65; gender: F; race: W; pmi: 18.5; ph: 6.5; rin: 7; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304874                                   age: 55; gender: M; race: W; pmi: 28; ph: 6.1; rin: 6.8; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304875                                 age: 22; gender: M; race: W; pmi: 20.1; ph: 6.8; rin: 7.1; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304876                                   age: 52; gender: F; race: W; pmi: 22.6; ph: 7.1; rin: 7; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304877                                 age: 58; gender: F; race: W; pmi: 22.7; ph: 6.4; rin: 6.3; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304878                                 age: 40; gender: F; race: B; pmi: 16.6; ph: 6.8; rin: 7.9; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304879                                 age: 41; gender: F; race: W; pmi: 15.4; ph: 6.6; rin: 8.5; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304880                                 age: 49; gender: M; race: W; pmi: 21.2; ph: 6.5; rin: 7.8; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304881                                age: 48; gender: M; race: W; pmi: 21.68; ph: 6.6; rin: 7.3; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304882                                 age: 39; gender: F; race: W; pmi: 24.5; ph: 6.8; rin: 8.2; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304883                                   age: 48; gender: M; race: W; pmi: 24.5; ph: 6.5; rin: 7; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304884                                 age: 43; gender: M; race: W; pmi: 13.8; ph: 6.6; rin: 7.6; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304885                                 age: 68; gender: M; race: W; pmi: 11.8; ph: 6.8; rin: 6.1; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304886                                 age: 58; gender: F; race: W; pmi: 18.8; ph: 6.6; rin: 7.2; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304887                                 age: 43; gender: M; race: W; pmi: 22.3; ph: 6.7; rin: 7.9; tissue: hippocampus; disease state: control</span></span>
<span><span class="co">#&gt; GSM1304888               age: 51; gender: M; race: W; pmi: 24.6; ph: 6.5; rin: 7.7; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304889               age: 53; gender: F; race: W; pmi: 11.9; ph: 6.7; rin: 8.1; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304890               age: 26; gender: F; race: W; pmi: 13.4; ph: 6.4; rin: 7.5; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304891               age: 52; gender: F; race: W; pmi: 10.3; ph: 6.5; rin: 6.6; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304892                 age: 62; gender: M; race: W; pmi: 26; ph: 6.5; rin: 6.8; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304893               age: 29; gender: M; race: W; pmi: 26.6; ph: 6.9; rin: 7.8; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304894               age: 49; gender: F; race: W; pmi: 23.4; ph: 6.4; rin: 6.2; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304895               age: 54; gender: F; race: W; pmi: 17.9; ph: 6.2; rin: 6.1; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304896               age: 28; gender: F; race: B; pmi: 24.8; ph: 6.6; rin: 8.2; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304897               age: 42; gender: M; race: W; pmi: 14.3; ph: 6.4; rin: 6.2; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304898               age: 44; gender: M; race: W; pmi: 19.3; ph: 6.5; rin: 6.3; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304899                 age: 40; gender: F; race: W; pmi: 22.2; ph: 6.6; rin: 8; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304900                 age: 47; gender: M; race: W; pmi: 24; ph: 6.6; rin: 5.5; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304901                 age: 59; gender: M; race: W; pmi: 13; ph: 6.6; rin: 7.2; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304902               age: 47; gender: F; race: W; pmi: 22.3; ph: 6.6; rin: 6.5; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304903               age: 34; gender: M; race: W; pmi: 24.4; ph: 6.6; rin: 8.4; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304904                 age: 51; gender: M; race: W; pmi: 28.3; ph: 7.3; rin: 7; tissue: hippocampus; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304905                          age: 49; gender: M; race: W; pmi: 21.5; ph: 5.97; rin: 6; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304906                       age: 47; gender: F; race: W; pmi: 14.37; ph: 6.35; rin: 6.3; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304907                        age: 25; gender: F; race: B; pmi: 20.1; ph: 6.73; rin: 5.6; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304908                        age: 62; gender: M; race: W; pmi: 22.7; ph: 7.14; rin: 6.3; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304909                        age: 44; gender: F; race: W; pmi: 24.5; ph: 6.63; rin: 7.8; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304910                        age: 46; gender: F; race: W; pmi: 23.8; ph: 6.61; rin: 6.9; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304911                          age: 50; gender: M; race: W; pmi: 11; ph: 6.23; rin: 7.2; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304912                        age: 46; gender: M; race: W; pmi: 15.8; ph: 6.19; rin: 6.2; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304913                        age: 41; gender: F; race: W; pmi: 20.1; ph: 6.27; rin: 6.7; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304914                        age: 47; gender: M; race: W; pmi: 28.9; ph: 6.58; rin: 6.7; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304915                        age: 37; gender: M; race: B; pmi: 5.98; ph: 6.07; rin: 6.4; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304916                         age: 58; gender: M; race: W; pmi: 7.7; ph: 6.22; rin: 6.7; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304917                         age: 44; gender: F; race: B; pmi: 18.7; ph: 6.2; rin: 6.4; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304918                        age: 38; gender: M; race: W; pmi: 28.8; ph: 6.56; rin: 6.6; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304919                        age: 52; gender: M; race: B; pmi: 27.1; ph: 6.68; rin: 6.3; tissue: hippocampus; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304920          age: 52; gender: M; race: W; pmi: 23.5; ph: 6.7; rin: 7.2; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304921          age: 50; gender: F; race: W; pmi: 11.7; ph: 6.4; rin: 8.6; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304922          age: 28; gender: F; race: W; pmi: 22.3; ph: 6.3; rin: 8.6; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304923            age: 55; gender: F; race: W; pmi: 17.5; ph: 6.4; rin: 8; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304924          age: 58; gender: M; race: W; pmi: 27.7; ph: 6.8; rin: 7.5; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304925          age: 28; gender: M; race: W; pmi: 27.4; ph: 6.2; rin: 7.9; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304926          age: 49; gender: F; race: W; pmi: 21.5; ph: 6.7; rin: 8.1; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304927          age: 56; gender: F; race: W; pmi: 24.5; ph: 6.1; rin: 6.9; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304928          age: 50; gender: M; race: W; pmi: 12.1; ph: 6.7; rin: 7.6; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304929          age: 40; gender: M; race: W; pmi: 18.5; ph: 6.4; rin: 7.9; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304930          age: 39; gender: F; race: W; pmi: 22.2; ph: 6.7; rin: 7.8; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304931          age: 45; gender: M; race: W; pmi: 27.2; ph: 7.1; rin: 7.3; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304932          age: 42; gender: M; race: W; pmi: 12.5; ph: 6.7; rin: 7.6; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304933           age: 65; gender: M; race: W; pmi: 8.9; ph: 6.7; rin: 6.9; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304934          age: 51; gender: F; race: W; pmi: 21.5; ph: 6.7; rin: 7.7; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304935          age: 39; gender: M; race: W; pmi: 24.2; ph: 6.6; rin: 7.3; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304936          age: 48; gender: M; race: W; pmi: 18.1; ph: 6.9; rin: 8.2; tissue: Pre-frontal cortex (BA46); disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304937                   age: 51; gender: M; race: W; pmi: 24.2; ph: 6.6; rin: 7.9; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304938                    age: 51; gender: F; race: W; pmi: 7.8; ph: 6.6; rin: 8.6; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304939                   age: 36; gender: F; race: W; pmi: 14.5; ph: 6.4; rin: 8.6; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304940                   age: 65; gender: F; race: W; pmi: 18.5; ph: 6.5; rin: 8.3; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304941                     age: 55; gender: M; race: W; pmi: 28; ph: 6.1; rin: 7.9; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304942                   age: 22; gender: M; race: W; pmi: 20.1; ph: 6.8; rin: 8.1; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304943                   age: 52; gender: F; race: W; pmi: 22.6; ph: 7.1; rin: 8.2; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304944                     age: 58; gender: F; race: W; pmi: 22.7; ph: 6.4; rin: 8; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304945                   age: 40; gender: F; race: B; pmi: 16.6; ph: 6.8; rin: 8.2; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304946                   age: 41; gender: F; race: W; pmi: 15.4; ph: 6.6; rin: 8.2; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304947                   age: 49; gender: M; race: W; pmi: 21.2; ph: 6.5; rin: 7.9; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304948                  age: 48; gender: M; race: W; pmi: 21.68; ph: 6.6; rin: 7.5; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304949                   age: 39; gender: F; race: W; pmi: 24.5; ph: 6.8; rin: 7.4; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304950                   age: 48; gender: M; race: W; pmi: 24.5; ph: 6.5; rin: 6.8; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304951                   age: 43; gender: M; race: W; pmi: 13.8; ph: 6.6; rin: 7.5; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304952                   age: 68; gender: M; race: W; pmi: 11.8; ph: 6.8; rin: 6.7; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304953                   age: 58; gender: F; race: W; pmi: 18.8; ph: 6.6; rin: 8.7; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304954                     age: 43; gender: M; race: W; pmi: 22.3; ph: 6.7; rin: 8; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304955                     age: 46; gender: M; race: W; pmi: 22; ph: 6.3; rin: 6.6; tissue: Pre-frontal cortex (BA46); disease state: control</span></span>
<span><span class="co">#&gt; GSM1304956 age: 51; gender: M; race: W; pmi: 24.6; ph: 6.5; rin: 7.8; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304957 age: 53; gender: F; race: W; pmi: 11.9; ph: 6.7; rin: 8.4; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304958 age: 26; gender: F; race: W; pmi: 13.4; ph: 6.4; rin: 8.4; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304959 age: 52; gender: F; race: W; pmi: 10.3; ph: 6.5; rin: 8.1; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304960   age: 62; gender: M; race: W; pmi: 26; ph: 6.5; rin: 7.8; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304961 age: 29; gender: M; race: W; pmi: 26.6; ph: 6.9; rin: 8.2; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304962 age: 49; gender: F; race: W; pmi: 23.4; ph: 6.4; rin: 7.6; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304963 age: 54; gender: F; race: W; pmi: 17.9; ph: 6.2; rin: 7.5; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304964 age: 28; gender: F; race: B; pmi: 24.8; ph: 6.6; rin: 7.9; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304965 age: 42; gender: M; race: W; pmi: 14.3; ph: 6.4; rin: 8.4; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304966 age: 40; gender: F; race: W; pmi: 22.2; ph: 6.6; rin: 7.7; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304967   age: 47; gender: M; race: W; pmi: 24; ph: 6.6; rin: 6.8; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304968   age: 44; gender: M; race: W; pmi: 11; ph: 6.5; rin: 7.2; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304969   age: 59; gender: M; race: W; pmi: 13; ph: 6.6; rin: 7.7; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304970 age: 47; gender: F; race: W; pmi: 22.3; ph: 6.6; rin: 6.7; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304971 age: 34; gender: M; race: W; pmi: 24.4; ph: 6.6; rin: 7.8; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304972 age: 51; gender: M; race: W; pmi: 28.3; ph: 7.3; rin: 7.7; tissue: Pre-frontal cortex (BA46); disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1304973            age: 49; gender: M; race: W; pmi: 21.5; ph: 5.97; rin: 7; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304974           age: 47; gender: F; race: W; pmi: 14.37; ph: 6.35; rin: 9; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304975          age: 25; gender: F; race: B; pmi: 20.1; ph: 6.73; rin: 7.2; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304976           age: 41; gender: F; race: W; pmi: 17.1; ph: 6.9; rin: 8.3; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304977          age: 62; gender: M; race: W; pmi: 22.7; ph: 7.14; rin: 8.1; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304978           age: 47; gender: F; race: B; pmi: 20.1; ph: 7.3; rin: 8.1; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304979          age: 44; gender: F; race: W; pmi: 24.5; ph: 6.63; rin: 7.6; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304980            age: 46; gender: F; race: W; pmi: 23.8; ph: 6.61; rin: 8; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304981            age: 50; gender: M; race: W; pmi: 11; ph: 6.23; rin: 8.4; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304982          age: 41; gender: F; race: W; pmi: 20.1; ph: 6.27; rin: 7.4; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304983            age: 47; gender: M; race: W; pmi: 28.9; ph: 6.58; rin: 7; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304984          age: 37; gender: M; race: B; pmi: 5.98; ph: 6.07; rin: 6.3; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304985           age: 58; gender: M; race: W; pmi: 7.7; ph: 6.22; rin: 7.3; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304986           age: 44; gender: F; race: B; pmi: 18.7; ph: 6.2; rin: 7.6; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304987          age: 52; gender: M; race: B; pmi: 27.1; ph: 6.68; rin: 7.4; tissue: Pre-frontal cortex (BA46); disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1304988               age: 50; gender: M; race: W; pmi: 12.1; ph: 6.7; rin: 8.6; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304989               age: 40; gender: M; race: W; pmi: 18.5; ph: 6.4; rin: 8.4; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304990               age: 39; gender: F; race: W; pmi: 22.2; ph: 6.7; rin: 9.1; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304991               age: 45; gender: M; race: W; pmi: 27.2; ph: 7.1; rin: 8.7; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304992               age: 42; gender: M; race: W; pmi: 12.5; ph: 6.7; rin: 8.7; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304993                age: 65; gender: M; race: W; pmi: 8.9; ph: 6.7; rin: 8.3; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304994               age: 51; gender: F; race: W; pmi: 21.5; ph: 6.7; rin: 8.4; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304995               age: 39; gender: M; race: W; pmi: 24.2; ph: 6.6; rin: 8.5; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304996               age: 48; gender: M; race: W; pmi: 18.1; ph: 6.9; rin: 8.8; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304997               age: 52; gender: M; race: W; pmi: 23.5; ph: 6.7; rin: 9.1; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304998               age: 50; gender: F; race: W; pmi: 11.7; ph: 6.4; rin: 8.4; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1304999                 age: 28; gender: F; race: W; pmi: 22.3; ph: 6.3; rin: 9; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305000                 age: 55; gender: F; race: W; pmi: 17.5; ph: 6.4; rin: 6; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305001               age: 58; gender: M; race: W; pmi: 27.7; ph: 6.8; rin: 6.6; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305002               age: 49; gender: F; race: W; pmi: 21.5; ph: 6.7; rin: 8.7; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305003               age: 56; gender: F; race: W; pmi: 24.5; ph: 6.1; rin: 7.7; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305004               age: 42; gender: F; race: W; pmi: 31.2; ph: 6.5; rin: 6.8; tissue: Associative striatum; disease state: bipolar disorder</span></span>
<span><span class="co">#&gt; GSM1305005                        age: 49; gender: M; race: W; pmi: 21.2; ph: 6.5; rin: 8.4; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305006                       age: 48; gender: M; race: W; pmi: 21.68; ph: 6.6; rin: 7.5; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305007                        age: 39; gender: F; race: W; pmi: 24.5; ph: 6.8; rin: 7.5; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305008                        age: 48; gender: M; race: W; pmi: 24.5; ph: 6.5; rin: 7.6; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305009                        age: 43; gender: M; race: W; pmi: 13.8; ph: 6.6; rin: 8.7; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305010                        age: 68; gender: M; race: W; pmi: 11.8; ph: 6.8; rin: 8.5; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305011                        age: 58; gender: F; race: W; pmi: 18.8; ph: 6.6; rin: 8.6; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305012                        age: 43; gender: M; race: W; pmi: 22.3; ph: 6.7; rin: 8.5; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305013                            age: 46; gender: M; race: W; pmi: 22; ph: 6.3; rin: 7; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305014                        age: 51; gender: M; race: W; pmi: 24.2; ph: 6.6; rin: 8.3; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305015                           age: 51; gender: F; race: W; pmi: 7.8; ph: 6.6; rin: 9; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305016                        age: 36; gender: F; race: W; pmi: 14.5; ph: 6.4; rin: 9.3; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305017                        age: 65; gender: F; race: W; pmi: 18.5; ph: 6.5; rin: 7.4; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305018                          age: 55; gender: M; race: W; pmi: 28; ph: 6.1; rin: 7.6; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305019                        age: 22; gender: M; race: W; pmi: 20.1; ph: 6.8; rin: 7.4; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305020                        age: 52; gender: F; race: W; pmi: 22.6; ph: 7.1; rin: 8.8; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305021                          age: 58; gender: F; race: W; pmi: 22.7; ph: 6.4; rin: 9; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305022                        age: 40; gender: F; race: B; pmi: 16.6; ph: 6.8; rin: 8.7; tissue: Associative striatum; disease state: control</span></span>
<span><span class="co">#&gt; GSM1305023      age: 42; gender: M; race: W; pmi: 14.3; ph: 6.4; rin: 8.7; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305024      age: 44; gender: M; race: W; pmi: 19.3; ph: 6.5; rin: 8.5; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305025        age: 47; gender: M; race: W; pmi: 24; ph: 6.6; rin: 7.3; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305026        age: 44; gender: M; race: W; pmi: 11; ph: 6.5; rin: 7.7; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305027        age: 59; gender: M; race: W; pmi: 13; ph: 6.6; rin: 8.4; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305028      age: 47; gender: F; race: W; pmi: 22.3; ph: 6.6; rin: 8.2; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305029      age: 34; gender: M; race: W; pmi: 24.4; ph: 6.6; rin: 9.1; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305030      age: 51; gender: M; race: W; pmi: 28.3; ph: 7.3; rin: 8.6; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305031      age: 51; gender: M; race: W; pmi: 24.6; ph: 6.5; rin: 8.3; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305032      age: 53; gender: F; race: W; pmi: 11.9; ph: 6.7; rin: 8.8; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305033      age: 26; gender: F; race: W; pmi: 13.4; ph: 6.4; rin: 9.2; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305034      age: 52; gender: F; race: W; pmi: 10.3; ph: 6.5; rin: 6.7; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305035        age: 62; gender: M; race: W; pmi: 26; ph: 6.5; rin: 7.5; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305036      age: 29; gender: M; race: W; pmi: 26.6; ph: 6.9; rin: 9.2; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305037      age: 49; gender: F; race: W; pmi: 23.4; ph: 6.4; rin: 6.7; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305038        age: 54; gender: F; race: W; pmi: 17.9; ph: 6.2; rin: 9; tissue: Associative striatum; disease state: major depressive disorder</span></span>
<span><span class="co">#&gt; GSM1305039                 age: 50; gender: M; race: W; pmi: 11; ph: 6.23; rin: 8.5; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305040               age: 46; gender: M; race: W; pmi: 15.8; ph: 6.19; rin: 7.8; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305041               age: 41; gender: F; race: W; pmi: 20.1; ph: 6.27; rin: 8.6; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305042               age: 47; gender: M; race: W; pmi: 28.9; ph: 6.58; rin: 8.4; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305043               age: 37; gender: M; race: B; pmi: 5.98; ph: 6.07; rin: 6.9; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305044                age: 58; gender: M; race: W; pmi: 7.7; ph: 6.22; rin: 6.7; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305045                age: 44; gender: F; race: B; pmi: 18.7; ph: 6.2; rin: 6.9; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305046               age: 38; gender: M; race: W; pmi: 28.8; ph: 6.56; rin: 6.8; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305047               age: 52; gender: M; race: B; pmi: 27.1; ph: 6.68; rin: 8.5; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305048               age: 49; gender: M; race: W; pmi: 21.5; ph: 5.97; rin: 8.4; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305049              age: 47; gender: F; race: W; pmi: 14.37; ph: 6.35; rin: 8.9; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305050               age: 25; gender: F; race: B; pmi: 20.1; ph: 6.73; rin: 7.3; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305051                age: 41; gender: F; race: W; pmi: 17.1; ph: 6.9; rin: 7.3; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305052               age: 62; gender: M; race: W; pmi: 22.7; ph: 7.14; rin: 7.8; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305053               age: 32; gender: M; race: W; pmi: 30.8; ph: 6.18; rin: 7.1; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305054                age: 47; gender: F; race: B; pmi: 20.1; ph: 7.3; rin: 8.8; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305055                 age: 50; gender: F; race: B; pmi: 22.9; ph: 6.25; rin: 8; tissue: Associative striatum; disease state: schizo; phrenia</span></span>
<span><span class="co">#&gt; GSM1305056                 age: 44; gender: F; race: W; pmi: 24.5; ph: 6.63; rin: 9; tissue: Associative striatum; disease state: schizo; phrenia</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="download-supplementary-data-from-geo-database---geo_suppl">Download supplementary data from GEO database - <code>geo_suppl</code>
<a class="anchor" aria-label="anchor" href="#download-supplementary-data-from-geo-database---geo_suppl"></a>
</h3>
<p>GEO stores raw data and processed sequence data files as the external supplementary data files. Sometimes, we may want to preprocess and normalize the rawdata by ourselves, in addition, it’s not uncommon that a GSE entity series matrix won’t contain the expression matrix, which is almost the case of high-throughout sequencing data. <code>geo_suppl</code> is designed for these conditions. Usually, the expression matrix will be provided in the GSE supplementary files or in the GSM supplementary files.</p>
<p>If the expression matrix is given in the GSE supplementary files, we can download it directly use <code>geo_suppl</code>, which will return a character vector containing the path of downloaded files.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse160724</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo_suppl.html">geo_suppl</a></span><span class="op">(</span></span>
<span>    ids <span class="op">=</span> <span class="st">"GSE160724"</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    pattern <span class="op">=</span> <span class="st">"counts_anno"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GSE suppl file from FTP site</span></span>
<span><span class="va">gse160724_dt</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">gse160724</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gse160724_dt</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;    gene_id  NC_1  NC_2 shSRSF1_1 shSRSF1_2</span></span>
<span><span class="co">#&gt;     &lt;char&gt; &lt;int&gt; &lt;int&gt;     &lt;int&gt;     &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1:    A1BG   189   179       299       310</span></span>
<span><span class="co">#&gt; 2:    A1CF     0     0         0         0</span></span>
<span><span class="co">#&gt; 3:     A2M     0     0         0         0</span></span>
<span><span class="co">#&gt; 4:   A2ML1     0     0         0         0</span></span>
<span><span class="co">#&gt; 5: A3GALT2     0     1         0         0</span></span>
<span><span class="co">#&gt;                                                          Dbxref</span></span>
<span><span class="co">#&gt;                                                          &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:          GeneID:1,Genbank:NM_130786.3,HGNC:HGNC:5,MIM:138670</span></span>
<span><span class="co">#&gt; 2:             GeneID:29974,Genbank:NM_138933.2,HGNC:HGNC:24086</span></span>
<span><span class="co">#&gt; 3:       GeneID:2,Genbank:NM_001347423.1,HGNC:HGNC:7,MIM:103950</span></span>
<span><span class="co">#&gt; 4: GeneID:144568,Genbank:NM_144670.5,HGNC:HGNC:23336,MIM:610627</span></span>
<span><span class="co">#&gt; 5:         GeneID:127550,Genbank:NM_001080438.1,HGNC:HGNC:30005</span></span>
<span><span class="co">#&gt;                              product</span></span>
<span><span class="co">#&gt;                               &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:            alpha-1-B glycoprotein</span></span>
<span><span class="co">#&gt; 2:    APOBEC1 complementation factor</span></span>
<span><span class="co">#&gt; 3:             alpha-2-macroglobulin</span></span>
<span><span class="co">#&gt; 4:      alpha-2-macroglobulin like 1</span></span>
<span><span class="co">#&gt; 5: alpha 1,3-galactosyltransferase 2</span></span>
<span><span class="co">#&gt;                                                                                                                       GO_id</span></span>
<span><span class="co">#&gt;                                                                                                                      &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:                                                                                                                         </span></span>
<span><span class="co">#&gt; 2:            GO:0003723,GO:0003727,GO:0005654,GO:0005737,GO:0005783,GO:0006397,GO:0016554,GO:0016556,GO:0030895,GO:0050821</span></span>
<span><span class="co">#&gt; 3:                                                                                                                         </span></span>
<span><span class="co">#&gt; 4:                                                                   GO:0004867,GO:0005615,GO:0030414,GO:0052548,GO:0070062</span></span>
<span><span class="co">#&gt; 5: GO:0005794,GO:0005975,GO:0006688,GO:0009247,GO:0016021,GO:0016757,GO:0030259,GO:0031982,GO:0032580,GO:0046872,GO:0047276</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                       GO_term</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                        &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:                                                                                                                                                                                                                                                                                                                                           </span></span>
<span><span class="co">#&gt; 2:                                                                                                                       RNA binding|single-stranded RNA binding|nucleoplasm|cytoplasm|endoplasmic reticulum|mRNA processing|cytidine to uridine editing|mRNA modification|apolipoprotein B mRNA editing enzyme complex|protein stabilization</span></span>
<span><span class="co">#&gt; 3:                                                                                                                                                                                                                                                                                                                                           </span></span>
<span><span class="co">#&gt; 4:                                                                                                                                                                                   serine-type endopeptidase inhibitor activity|extracellular space|peptidase inhibitor activity|regulation of endopeptidase activity|extracellular exosome</span></span>
<span><span class="co">#&gt; 5: Golgi apparatus|carbohydrate metabolic process|glycosphingolipid biosynthetic process|glycolipid biosynthetic process|integral component of membrane|transferase activity, transferring glycosyl groups|lipid glycosylation|vesicle|Golgi cisterna membrane|metal ion binding|N-acetyllactosaminide 3-alpha-galactosyltransferase activity</span></span>
<span><span class="co">#&gt;     pathway                                        pathway_description</span></span>
<span><span class="co">#&gt;      &lt;char&gt;                                                     &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:                                                                    </span></span>
<span><span class="co">#&gt; 2:                                                                    </span></span>
<span><span class="co">#&gt; 3: hsa04610                        Complement and coagulation cascades</span></span>
<span><span class="co">#&gt; 4:                                                                    </span></span>
<span><span class="co">#&gt; 5: hsa00603 Glycosphingolipid biosynthesis - globo and isoglobo series</span></span></code></pre></div>
<p>If the expression matrix is given in the GSM supplementary files, in this way, we start from derive all GSM accession ids and then download all GSM supplementary files and combine them into a expression matrix. Although no expression matrix in the series matrix file, it still contains the samples informations.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse180383_smat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span></span>
<span>    <span class="st">"GSE180383"</span>,</span>
<span>    odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    gse_matrix <span class="op">=</span> <span class="cn">TRUE</span>, add_gpl <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    pdata_from_soft <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GSE matrix file from FTP site</span></span>
<span><span class="co">#&gt; → Parsing 1 series matrix file of GSE180383</span></span>
<span><span class="co">#&gt; Warning: Cannot parse characteristic column correctly</span></span>
<span><span class="co">#&gt; ℹ Details see "characteristics_ch1" column in phenoData</span></span>
<span><span class="co">#&gt; ℹ Please use `parse_pdata()` or `parse_gsm_list()` function to convert it</span></span>
<span><span class="co">#&gt;   manually if necessary!</span></span>
<span><span class="co">#&gt; ✔ Parsing 1 GSE series matrix successfully!</span></span>
<span><span class="co">#&gt; → Constructing &lt;ExpressionSet&gt;</span></span>
<span><span class="co">#&gt; ✔ Found Bioconductor annotation package for "GPL21359"</span></span>
<span><span class="va">gse180383_smat_cli</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse180383_smat</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gse180383_smat_cli</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                  title geo_accession                status</span></span>
<span><span class="co">#&gt; GSM5461787 Monoecious WT RNA-seq  rep1    GSM5461787 Public on Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461788 Monoecious WT RNA-seq  rep2    GSM5461788 Public on Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461789 Monoecious WT RNA-seq  rep3    GSM5461789 Public on Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461790        Cmlhp1abRNA-seq rep1    GSM5461790 Public on Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461791        Cmlhp1abRNA-seq rep2    GSM5461791 Public on Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461792        Cmlhp1abRNA-seq rep3    GSM5461792 Public on Feb 15 2022</span></span>
<span><span class="co">#&gt;            submission_date last_update_date</span></span>
<span><span class="co">#&gt; GSM5461787     Jul 19 2021      Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461788     Jul 19 2021      Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461789     Jul 19 2021      Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461790     Jul 19 2021      Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461791     Jul 19 2021      Feb 15 2022</span></span>
<span><span class="co">#&gt; GSM5461792     Jul 19 2021      Feb 15 2022</span></span>
<span><span class="va">gse180383_smat_gsmids</span> <span class="op">&lt;-</span> <span class="va">gse180383_smat_cli</span><span class="op">[[</span><span class="st">"geo_accession"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">gse180383_smat_gsm_suppl</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo_suppl.html">geo_suppl</a></span><span class="op">(</span><span class="va">gse180383_smat_gsmids</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 6 GSM suppl files from FTP site</span></span></code></pre></div>
<p>Another way, we can also derive sample accession ids from GSE soft files, which is what our laboratory prefers to since we can easily get exact sample traits information as described in the above by utilizing <code>parse_gsm_list</code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse180383_soft</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo.html">geo</a></span><span class="op">(</span></span>
<span>    <span class="st">"GSE180383"</span>,</span>
<span>    odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    gse_matrix <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Downloading 1 GSE soft file from FTP site</span></span>
<span><span class="va">gse180383_soft_cli</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/parse_gsm_list.html">parse_gsm_list</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">gsm</a></span><span class="op">(</span><span class="va">gse180383_soft</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: More than one characters ":" found in meta characteristics data</span></span>
<span><span class="co">#&gt; ℹ Details see: "characteristics_ch1" column in returned data.</span></span>
<span><span class="co">#&gt; ℹ Please use `parse_pdata()` or combine `strsplit()` and `parse_gsm_list()`</span></span>
<span><span class="co">#&gt;   function to convert it manually if necessary!</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gse180383_soft_cli</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;            channel_count</span></span>
<span><span class="co">#&gt; GSM5461787             1</span></span>
<span><span class="co">#&gt; GSM5461788             1</span></span>
<span><span class="co">#&gt; GSM5461789             1</span></span>
<span><span class="co">#&gt; GSM5461790             1</span></span>
<span><span class="co">#&gt; GSM5461791             1</span></span>
<span><span class="co">#&gt; GSM5461792             1</span></span>
<span><span class="co">#&gt;                                                              ch1_cultivar</span></span>
<span><span class="co">#&gt; GSM5461787 Charantais type: Cucumis melo L. subsp. melo var cantalupensis</span></span>
<span><span class="co">#&gt; GSM5461788 Charantais type: Cucumis melo L. subsp. melo var cantalupensis</span></span>
<span><span class="co">#&gt; GSM5461789 Charantais type: Cucumis melo L. subsp. melo var cantalupensis</span></span>
<span><span class="co">#&gt; GSM5461790 Charantais type: Cucumis melo L. subsp. melo var cantalupensis</span></span>
<span><span class="co">#&gt; GSM5461791 Charantais type: Cucumis melo L. subsp. melo var cantalupensis</span></span>
<span><span class="co">#&gt; GSM5461792 Charantais type: Cucumis melo L. subsp. melo var cantalupensis</span></span>
<span><span class="co">#&gt;                                                                                                                                                      ch1_genotypes</span></span>
<span><span class="co">#&gt; GSM5461787                                                                                                                                   CharMONO inbreed line</span></span>
<span><span class="co">#&gt; GSM5461788                                                                                                                                   CharMONO inbreed line</span></span>
<span><span class="co">#&gt; GSM5461789                                                                                                                                   CharMONO inbreed line</span></span>
<span><span class="co">#&gt; GSM5461790 CharMONO cmlhp1ab double mutant carrying EMS mutations for Cmlhp1a (G1970A, genomic position from ATG ) and cmlhp1b (C1930T genomic position from ATG )</span></span>
<span><span class="co">#&gt; GSM5461791 CharMONO cmlhp1ab double mutant carrying EMS mutations for Cmlhp1a (G1970A, genomic position from ATG ) and cmlhp1b (C1930T genomic position from ATG )</span></span>
<span><span class="co">#&gt; GSM5461792 CharMONO cmlhp1ab double mutant carrying EMS mutations for Cmlhp1a (G1970A, genomic position from ATG ) and cmlhp1b (C1930T genomic position from ATG )</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                     characteristics_ch1</span></span>
<span><span class="co">#&gt; GSM5461787                                                                                                                                   cultivar: Charantais type: Cucumis melo L. subsp. melo var cantalupensis; genotypes: CharMONO inbreed line</span></span>
<span><span class="co">#&gt; GSM5461788                                                                                                                                   cultivar: Charantais type: Cucumis melo L. subsp. melo var cantalupensis; genotypes: CharMONO inbreed line</span></span>
<span><span class="co">#&gt; GSM5461789                                                                                                                                   cultivar: Charantais type: Cucumis melo L. subsp. melo var cantalupensis; genotypes: CharMONO inbreed line</span></span>
<span><span class="co">#&gt; GSM5461790 cultivar: Charantais type: Cucumis melo L. subsp. melo var cantalupensis; genotypes: CharMONO cmlhp1ab double mutant carrying EMS mutations for Cmlhp1a (G1970A, genomic position from ATG ) and cmlhp1b (C1930T genomic position from ATG )</span></span>
<span><span class="co">#&gt; GSM5461791 cultivar: Charantais type: Cucumis melo L. subsp. melo var cantalupensis; genotypes: CharMONO cmlhp1ab double mutant carrying EMS mutations for Cmlhp1a (G1970A, genomic position from ATG ) and cmlhp1b (C1930T genomic position from ATG )</span></span>
<span><span class="co">#&gt; GSM5461792 cultivar: Charantais type: Cucumis melo L. subsp. melo var cantalupensis; genotypes: CharMONO cmlhp1ab double mutant carrying EMS mutations for Cmlhp1a (G1970A, genomic position from ATG ) and cmlhp1b (C1930T genomic position from ATG )</span></span>
<span><span class="co">#&gt;             contact_address</span></span>
<span><span class="co">#&gt; GSM5461787 630 Rue Noetzlin</span></span>
<span><span class="co">#&gt; GSM5461788 630 Rue Noetzlin</span></span>
<span><span class="co">#&gt; GSM5461789 630 Rue Noetzlin</span></span>
<span><span class="co">#&gt; GSM5461790 630 Rue Noetzlin</span></span>
<span><span class="co">#&gt; GSM5461791 630 Rue Noetzlin</span></span>
<span><span class="co">#&gt; GSM5461792 630 Rue Noetzlin</span></span>
<span><span class="va">gse180383_soft_gsmids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="reference/GEO-class.html">gsm</a></span><span class="op">(</span><span class="va">gse180383_soft</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gse180383_soft_gsm_suppl</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/geo_suppl.html">geo_suppl</a></span><span class="op">(</span><span class="va">gse180383_soft_gsmids</span>, odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Finding 6 {.strong GSM} {.field suppl} file already downloaded:</span></span>
<span><span class="co">#&gt; 'GSM5461787_trim_RNA_Mono_1_S13_R1_001_countsMatrix.txt.gz',</span></span>
<span><span class="co">#&gt; 'GSM5461788_trim_RNA_Mono_2_S14_R1_001_countsMatrix.txt.gz',</span></span>
<span><span class="co">#&gt; 'GSM5461789_trim_RNA_Mono_3_S15_R1_001_countsMatrix.txt.gz',</span></span>
<span><span class="co">#&gt; 'GSM5461790_trim_RNA_ab_1_S16_R1_001_countsMatrix.txt.gz',</span></span>
<span><span class="co">#&gt; 'GSM5461791_trim_RNA_ab_2_S17_R1_001_countsMatrix.txt.gz', and</span></span>
<span><span class="co">#&gt; 'GSM5461792_trim_RNA_ab_3_S18_R1_001_countsMatrix.txt.gz'</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="other-utilities">Other utilities<a class="anchor" aria-label="anchor" href="#other-utilities"></a>
</h3>
<p><code>geokit</code> also provide some useful function to help better interact with GEO.</p>
<ul>
<li>
<code>geo_show</code> function: Require a geo entity id and open GEO Accession site in the default browser.</li>
<li>
<code>log_trans</code> function: Require a expression matrix and this function will check whether this expression matrix has experienced logarithmic transformation, if it hasn’t, <code>log_trans</code> will do it. This is a helper function used in <code>GEO2R</code>.</li>
</ul>
</div>
<div class="section level3">
<h3 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/libmkl_rt.so;  LAPACK version 3.8.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Shanghai</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] geokit_0.0.1.9000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] vctrs_0.6.5        cli_3.6.5          knitr_1.50         rlang_1.1.6       </span></span>
<span><span class="co">#&gt;  [5] xfun_0.52          stringi_1.8.4      generics_0.1.3     textshaping_0.4.0 </span></span>
<span><span class="co">#&gt;  [9] data.table_1.16.99 glue_1.8.0         htmltools_0.5.8.1  ragg_1.3.3        </span></span>
<span><span class="co">#&gt; [13] fansi_1.0.6        rmarkdown_2.29     evaluate_1.0.3     tibble_3.2.1      </span></span>
<span><span class="co">#&gt; [17] fastmap_1.2.0      yaml_2.3.10        lifecycle_1.0.4    stringr_1.5.1     </span></span>
<span><span class="co">#&gt; [21] compiler_4.4.2     dplyr_1.1.4        pkgconfig_2.0.3    systemfonts_1.1.0 </span></span>
<span><span class="co">#&gt; [25] digest_0.6.37      R6_2.6.1           tidyselect_1.2.1   utf8_1.2.5        </span></span>
<span><span class="co">#&gt; [29] pillar_1.9.0       magrittr_2.0.3     withr_3.0.2        tools_4.4.2</span></span></code></pre></div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Yunuuuu/geokit/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/Yunuuuu/geokit/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing geokit</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Yun Peng <br><small class="roles"> Author, maintainer </small>   </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Yunuuuu/geokit/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/Yunuuuu/geokit/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/Yunuuuu/geokit" class="external-link"><img src="https://codecov.io/gh/Yunuuuu/geokit/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yun Peng.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
