<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Parse key-value pairs in GEO series soft file — parse_gsm_list • geokit</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Parse key-value pairs in GEO series soft file — parse_gsm_list"><meta name="description" content='Lots of GSEs now use "characteristics_ch*" meta header data for key-value
pairs of annotation. If that is the case, this simply cleans the GSM
GEOSoft @meta slot up and transforms the keys to column names and the
values to column values.'><meta property="og:description" content='Lots of GSEs now use "characteristics_ch*" meta header data for key-value
pairs of annotation. If that is the case, this simply cleans the GSM
GEOSoft @meta slot up and transforms the keys to column names and the
values to column values.'></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">geokit</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Yunuuuu/geokit/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Parse key-value pairs in GEO series soft file</h1>
      <small class="dont-index">Source: <a href="https://github.com/Yunuuuu/geokit/blob/main/R/phenodata.R" class="external-link"><code>R/phenodata.R</code></a></small>
      <div class="d-none name"><code>parse_gsm_list.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Lots of GSEs now use <code>"characteristics_ch*"</code> meta header data for key-value
pairs of annotation. If that is the case, this simply cleans the GSM
<code>GEOSoft</code> @meta slot up and transforms the keys to column names and the
values to column values.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">parse_gsm_list</span><span class="op">(</span><span class="va">gsm_list</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gsm-list">gsm_list<a class="anchor" aria-label="anchor" href="#arg-gsm-list"></a></dt>
<dd><p>A list of GEOSoft, especially for @gsm slot in a <code>GEOSeries</code>
object.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a data.frame whose rows are samples and columns are the sample infos</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gse201530_soft</span> <span class="op">&lt;-</span> <span class="fu"><a href="geo.html">geo</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"GSE201530"</span>,</span></span>
<span class="r-in"><span>    odir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    gse_matrix <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading <span style="color: #0000BB;">1</span> <span style="font-weight: bold;">GSE</span> <span style="color: #00BB00;">soft</span> file from FTP site</span>
<span class="r-in"><span><span class="fu">parse_gsm_list</span><span class="op">(</span><span class="fu"><a href="GEO-class.html">gsm</a></span><span class="op">(</span><span class="va">gse201530_soft</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            channel_count ch1_gender ch1_age</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090             1       Male      58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091             1     Female      56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092             1     Female      37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093             1     Female      19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094             1     Female      31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095             1       Male      36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096             1     Female      21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097             1     Female      23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098             1       Male      50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099             1       Male      36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100             1     Female      30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101             1     Female      17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102             1       Male      20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103             1       Male      57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104             1       Male      30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105             1     Female      21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106             1       Male      23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107             1     Female      27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108             1     Female      80</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109             1     Female      38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110             1       Male      39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111             1     Female      29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112             1     Female      55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113             1       Male      26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114             1       Male      83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115             1     Female      23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116             1     Female      20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117             1       Male      82</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118             1     Female      51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119             1       Male      54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120             1     Female      83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121             1     Female      32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122             1       Male      36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123             1     Female      30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124             1     Female      28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125             1       Male      29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126             1       Male      61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127             1     Female      59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128             1     Female      54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129             1       Male      63</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130             1     Female      62</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131             1       Male      26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132             1     Female      24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133             1     Female      32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134             1     Female      35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135             1     Female      38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136             1     Female      37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137             1     Female      24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138             1     Female      40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139             1     Female      61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140             1     Female      42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141             1     Female      39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142             1     Female      39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143             1     Female      40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144             1       Male      28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            ch1_group (by covid-19 vaccination, prior infection)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097                                      Healthy control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110                      No vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128                      No vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129                      No vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131                      No vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132                      No vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136                      No vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137                         Vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140                      Vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143                   No vaccination, No prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144                      No vaccination, Prior infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            ch1_omicron sublineage ch1_days after positive pcr results</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097                     --                                  --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106                   BA.1                               day 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109                   BA.1                               day 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112                   BA.1                               day 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114                   BA.1                               day 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117                   BA.1                               day 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122                   BA.1                               day 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123                   BA.1                               day 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124                   BA.1                               day 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125                   BA.1                               day 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128                   BA.1                               day 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129                   BA.1                               day 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130                   BA.1                               day 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131                   BA.1                               day 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133                   BA.1                               day 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136                   BA.1                               day 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137                   BA.1                               day 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140                   BA.1                               day 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141                   BA.1                               day 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142                   BA.1                               day 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143                   BA.1                               day 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144                   BA.1                               day 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            ch1_disease state ch1_geographical location ch1_cell type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097   Healthy control                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144  COVID-19_Omicron                   Austria          PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                          characteristics_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090                            gender: Male; age: 58; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091                          gender: Female; age: 56; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092                          gender: Female; age: 37; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093                          gender: Female; age: 19; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094                          gender: Female; age: 31; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095                            gender: Male; age: 36; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096                          gender: Female; age: 21; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097                          gender: Female; age: 23; group (by covid-19 vaccination, prior infection): Healthy control; omicron sublineage: --; days after positive pcr results: --; disease state: Healthy control; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098      gender: Male; age: 50; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099      gender: Male; age: 36; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100    gender: Female; age: 30; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101    gender: Female; age: 17; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102      gender: Male; age: 20; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103      gender: Male; age: 57; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104      gender: Male; age: 30; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105    gender: Female; age: 21; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106      gender: Male; age: 23; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 3; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107    gender: Female; age: 27; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108    gender: Female; age: 80; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109    gender: Female; age: 38; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 3; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110      gender: Male; age: 39; group (by covid-19 vaccination, prior infection): No vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111    gender: Female; age: 29; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112    gender: Female; age: 55; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 3; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113      gender: Male; age: 26; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114   gender: Male; age: 83; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 1; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115    gender: Female; age: 23; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116    gender: Female; age: 20; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117      gender: Male; age: 82; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 1; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 gender: Female; age: 51; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119   gender: Male; age: 54; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 gender: Female; age: 83; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 gender: Female; age: 32; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122   gender: Male; age: 36; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 3; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 gender: Female; age: 30; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 4; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 gender: Female; age: 28; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 1; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125   gender: Male; age: 29; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 1; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126   gender: Male; age: 61; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 gender: Female; age: 59; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128    gender: Female; age: 54; group (by covid-19 vaccination, prior infection): No vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 5; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129      gender: Male; age: 63; group (by covid-19 vaccination, prior infection): No vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 5; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 gender: Female; age: 62; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 4; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131      gender: Male; age: 26; group (by covid-19 vaccination, prior infection): No vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 3; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132    gender: Female; age: 24; group (by covid-19 vaccination, prior infection): No vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 gender: Female; age: 32; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 1; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 gender: Female; age: 35; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 gender: Female; age: 38; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136    gender: Female; age: 37; group (by covid-19 vaccination, prior infection): No vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 1; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137       gender: Female; age: 24; group (by covid-19 vaccination, prior infection): Vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 5; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 gender: Female; age: 40; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 gender: Female; age: 61; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140    gender: Female; age: 42; group (by covid-19 vaccination, prior infection): Vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 1; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 gender: Female; age: 39; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 0; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 gender: Female; age: 39; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 3; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 gender: Female; age: 40; group (by covid-19 vaccination, prior infection): No vaccination, No prior infection; omicron sublineage: BA.1; days after positive pcr results: day 3; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144      gender: Male; age: 28; group (by covid-19 vaccination, prior infection): No vaccination, Prior infection; omicron sublineage: BA.1; days after positive pcr results: day 2; disease state: COVID-19_Omicron; geographical location: Austria; cell type: PBMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               contact_address contact_city contact_country</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 8 CENTER DR RM 107     Bethesda             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                         contact_department</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 National Institute of Diabetes and Digestive and Kidney (NIDDK)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   contact_email                   contact_institute</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 hyekyung.lee@nih.gov National Institutes of Health (NIH)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               contact_laboratory   contact_name contact_phone</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 Laboratory of Genetics and Physiology Hye Kyung,,Lee  301-435-6635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            contact_state contact_zip/postal_code</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144            MD                   20892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           data_processing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 RNA-seq data were trimmed using trimmomatic (Bolger et al., Bioinformatics, 2014) and subsequently aligned to the reference genome hg19 applying STAR RNA-seq aligner (Dobin et al., Bioinformatics, 2013). HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; HTSeq(Anders et al., Bioinformatics, 2015) was applied for obtain HTSeq counts.; Trimmed RNA-seq data were aligned to the human V, D and J genes and obtained immune clonotypes using MiXCR (Bolotin et al., Nat Methods, 2015; Bolotin et al., Nat Biotechnol, 2017).; Assembly: hg19; Supplementary files format and content: HTSeq counts from RNA-seq samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            data_row_count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143              0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                         extract_protocol_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 The buffy coat was extracted by centrifuge after blood sample collection and RBC was removed using RBC lysis buffer in the buffy coat. Total RNA was isolated using a Maxwell RSC simply RNA Blood purification kit.; Libraries for sequencing were prepared with TruSeq Stranded mRNA Kit following standard Illumina protocols.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            geo_accession</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090    GSM6066090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091    GSM6066091</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092    GSM6066092</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093    GSM6066093</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094    GSM6066094</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095    GSM6066095</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096    GSM6066096</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097    GSM6066097</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098    GSM6066098</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099    GSM6066099</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100    GSM6066100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101    GSM6066101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102    GSM6066102</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103    GSM6066103</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104    GSM6066104</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105    GSM6066105</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106    GSM6066106</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107    GSM6066107</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108    GSM6066108</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109    GSM6066109</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110    GSM6066110</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111    GSM6066111</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112    GSM6066112</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113    GSM6066113</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114    GSM6066114</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115    GSM6066115</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116    GSM6066116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117    GSM6066117</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118    GSM6066118</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119    GSM6066119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120    GSM6066120</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121    GSM6066121</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122    GSM6066122</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123    GSM6066123</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124    GSM6066124</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125    GSM6066125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126    GSM6066126</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127    GSM6066127</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128    GSM6066128</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129    GSM6066129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130    GSM6066130</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131    GSM6066131</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132    GSM6066132</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133    GSM6066133</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134    GSM6066134</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135    GSM6066135</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136    GSM6066136</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137    GSM6066137</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138    GSM6066138</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139    GSM6066139</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140    GSM6066140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141    GSM6066141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142    GSM6066142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143    GSM6066143</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144    GSM6066144</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                             growth_protocol_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 Blood samples were collected from the COVID-19 patients infected by Omicron varient.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 instrument_model last_update_date library_selection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 Illumina NovaSeq 6000      Jun 11 2022              cDNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            library_source library_strategy molecule_ch1 organism_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 transcriptomic          RNA-Seq    polyA RNA Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            platform_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144    GPL24676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                          relation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779606; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000698</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779605; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000699</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779604; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000700</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779603; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000701</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779602; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000702</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779601; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000703</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779600; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000704</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779599; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000705</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779598; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000706</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779597; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000707</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779596; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000708</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779595; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000709</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779594; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000710</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779593; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000711</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779592; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000712</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779591; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000713</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779590; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000714</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779589; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000715</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779588; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000716</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779587; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000717</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779586; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000718</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779585; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000719</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779584; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000720</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779583; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000721</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779582; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000722</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779581; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000723</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779580; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000724</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779579; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000725</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779578; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000726</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779577; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000727</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779576; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000728</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779575; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000729</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779574; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000730</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779573; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000731</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779572; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000732</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779571; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000733</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779570; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000734</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779569; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000735</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779568; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000736</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779567; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000737</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779566; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779565; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000739</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779564; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000740</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779563; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000741</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779562; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779561; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000743</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779560; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000744</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779559; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000745</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779558; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000746</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779557; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000747</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779556; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000748</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779555; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000749</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779554; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779553; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000751</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN27779552; SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX15000752</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            series_id source_name_ch1                status submission_date</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 GSE201530           PBMCs Public on Jun 11 2022     Apr 26 2022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                           supplementary_file_1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066090/suppl/GSM6066090_Naive_17_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066091/suppl/GSM6066091_Naive_18_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066092/suppl/GSM6066092_Naive_24_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066093/suppl/GSM6066093_Naive_26_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066094/suppl/GSM6066094_Naive_27_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066095/suppl/GSM6066095_Naive_28_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066096/suppl/GSM6066096_Naive_29_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066097/suppl/GSM6066097_Naive_30_Day0.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066098/suppl/GSM6066098_Omicron_1_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066099/suppl/GSM6066099_Omicron_2_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066100/suppl/GSM6066100_Omicron_3_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066101/suppl/GSM6066101_Omicron_4_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066102/suppl/GSM6066102_Omicron_5_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066103/suppl/GSM6066103_Omicron_6_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066104/suppl/GSM6066104_Omicron_7_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066105/suppl/GSM6066105_Omicron_8_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066106/suppl/GSM6066106_Omicron_9_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066107/suppl/GSM6066107_Omicron_10_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066108/suppl/GSM6066108_Omicron_11_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066109/suppl/GSM6066109_Omicron_12_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066110/suppl/GSM6066110_Omicron_13_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066111/suppl/GSM6066111_Omicron_14_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066112/suppl/GSM6066112_Omicron_15_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066113/suppl/GSM6066113_Omicron_16_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066114/suppl/GSM6066114_Omicron_17_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066115/suppl/GSM6066115_Omicron_18_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066116/suppl/GSM6066116_Omicron_19_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066117/suppl/GSM6066117_Omicron_20_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066118/suppl/GSM6066118_Omicron_24_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066119/suppl/GSM6066119_Omicron_25_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066120/suppl/GSM6066120_Omicron_26_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066121/suppl/GSM6066121_Omicron_27_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066122/suppl/GSM6066122_Omicron_28_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066123/suppl/GSM6066123_Omicron_29_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066124/suppl/GSM6066124_Omicron_30_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066125/suppl/GSM6066125_Omicron_31_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066126/suppl/GSM6066126_Omicron_33_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066127/suppl/GSM6066127_Omicron_34_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066128/suppl/GSM6066128_Omicron_35_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066129/suppl/GSM6066129_Omicron_37_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066130/suppl/GSM6066130_Omicron_38_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066131/suppl/GSM6066131_Omicron_42_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066132/suppl/GSM6066132_Omicron_43_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066133/suppl/GSM6066133_Omicron_48_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066134/suppl/GSM6066134_Omicron_49_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066135/suppl/GSM6066135_Omicron_50_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066136/suppl/GSM6066136_Omicron_51_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066137/suppl/GSM6066137_Omicron_52_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066138/suppl/GSM6066138_Omicron_53_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066139/suppl/GSM6066139_Omicron_55_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066140/suppl/GSM6066140_Omicron_65_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066141/suppl/GSM6066141_Omicron_66_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066142/suppl/GSM6066142_Omicron_67_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066143/suppl/GSM6066143_Omicron_68_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM6066nnn/GSM6066144/suppl/GSM6066144_Omicron_70_1st.txt.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            taxid_ch1          title type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066090      9606  Naive_17_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066091      9606  Naive_18_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066092      9606  Naive_24_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066093      9606  Naive_26_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066094      9606  Naive_27_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066095      9606  Naive_28_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066096      9606  Naive_29_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066097      9606  Naive_30_Day0  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066098      9606  Omicron_1_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066099      9606  Omicron_2_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066100      9606  Omicron_3_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066101      9606  Omicron_4_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066102      9606  Omicron_5_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066103      9606  Omicron_6_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066104      9606  Omicron_7_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066105      9606  Omicron_8_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066106      9606  Omicron_9_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066107      9606 Omicron_10_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066108      9606 Omicron_11_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066109      9606 Omicron_12_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066110      9606 Omicron_13_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066111      9606 Omicron_14_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066112      9606 Omicron_15_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066113      9606 Omicron_16_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066114      9606 Omicron_17_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066115      9606 Omicron_18_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066116      9606 Omicron_19_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066117      9606 Omicron_20_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066118      9606 Omicron_24_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066119      9606 Omicron_25_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066120      9606 Omicron_26_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066121      9606 Omicron_27_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066122      9606 Omicron_28_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066123      9606 Omicron_29_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066124      9606 Omicron_30_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066125      9606 Omicron_31_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066126      9606 Omicron_33_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066127      9606 Omicron_34_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066128      9606 Omicron_35_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066129      9606 Omicron_37_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066130      9606 Omicron_38_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066131      9606 Omicron_42_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066132      9606 Omicron_43_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066133      9606 Omicron_48_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066134      9606 Omicron_49_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066135      9606 Omicron_50_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066136      9606 Omicron_51_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066137      9606 Omicron_52_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066138      9606 Omicron_53_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066139      9606 Omicron_55_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066140      9606 Omicron_65_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066141      9606 Omicron_66_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066142      9606 Omicron_67_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066143      9606 Omicron_68_1st  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM6066144      9606 Omicron_70_1st  SRA</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yun Peng.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

